# -*- org-attach-id-dir: "~/src/personal/edmundmiller-dev/static/org-attach/" -*-
#+title: Nascent Transcript Identification
#+AUTHOR:    Edmund Miller
#+EMAIL:     Edmund.Miller@utdallas.edu
#+DATE:      2022-04-11 Mon
#+reveal_theme: white
#+options: num:nil
#+options: toc:nil
#+options: reveal_title_slide:nil
#+startup: inlineimages
#+startup: beamer
#+LaTeX_CLASS: beamer
#+LaTeX_CLASS_OPTIONS: [bigger]
#+BEAMER_FRAME_LEVEL: 2
# SPC m e l O

* Background
* Gene Regulation
** Central Dogma of Biology :ATTACH:
:PROPERTIES:
:ID:       b8ce871b-5b7f-4cef-b389-7b27459818b3
:END:


[[attachment:_20220407_195627screenshot.png]]

** A Simple View of Gene Expression :ATTACH:
:PROPERTIES:
:ID:       a9ae81d7-3773-4daa-baf6-bec17b6bb120
:BEAMER_env: column
:END:

*** Gene Expression :B_ignoreheading:BMCOL:
:PROPERTIES:
:BEAMER_col: 0.45
:BEAMER_env: block
:END:

[[attachment:_20220407_195540screenshot.png]]


*** Central Dogma :B_ignoreheading:BMCOL:
:PROPERTIES:
:BEAMER_col: 0.45
:BEAMER_env: block
:END:
[[attachment:_20220407_195803screenshot.png]]


** Enhancers

- Cis-acting DNA sequences that can increase the transcription of genes [fn:1:
  [cite:@Pennacchio_2013]]
- The human genome contains hundreds of thousands of enhancers
- Thought to work through DNA looping
- Evidence of Enhancer-Promoter interaction from cross-linking assays(3c)

** Topologically Associating Domain (TAD) :ATTACH:
:PROPERTIES:
:ID:       5dfc11b2-2dc8-4e7d-9034-57f92ff60114
:END:

#+caption: General topography of an active enhancer [cite:@banerji_1981].
[[attachment:_20220408_132400screenshot.png]]

** Enhancer-promoter Looping :ATTACH:
:PROPERTIES:
:ID:       1c40bd96-7754-4e2b-9a2b-028570f5d89b
:END:

#+caption: [cite:@Kraus_2017]
[[attachment:_20220408_144930screenshot.png]]



** Why are Enhancers difficult to identify?

1. Scattered across the 98% of the human genome that does not encode proteins [fn:1: [cite:@Pennacchio_2013]]
2. Enhancers location relative to their target gene (or genes) is highly
  variable. They can be upstream, downstream, or within introns. [fn:1:
  [cite:@Pennacchio_2013]]
3. Enhancers do not necessarily act on the respective closest promoter but can
  bypass neighbouring genes to regulate genes located more distantly along a
  chromosome [fn:1: [cite:@Pennacchio_2013]]

** Why are Enhancers difficult to identify?

4. One Enhancer can regulate multiple genes [fn:2: [cite:@Locksley_2001]] and One
  gene can be regulated by multiple enhancers [fn:3: [cite:@Kim_2018]]
5. The general sequence code of enhancers, if one exists at all, is poorly
  understood. [fn:1: [cite:@Pennacchio_2013]]
6. The activity of enhancers can be restricted to a particular tissue or cell type

** Multiple Enhancers can regulate one gene :ATTACH:
:PROPERTIES:
:ID:       41914259-ccb3-42b6-a38e-7e284c0bdded
:END:

#+caption[Short caption]: [cite:@Kim_2018]
[[attachment:_20220408_094258screenshot.png]]


** eRNAs Introduction

- Sense and antisense transcripts from enhancers
- Play a role in the activation because the knockdown of a subset of the eRNAs
  resulted in decreased gene transcription [fn:: [cite:@Shiekhattar_2010]]
- Used to identify active enhancers

* Global Transcriptional Activity Dynamics Reveal Functional Enhancer RNAs
** Global Transcriptional Activity Dynamics Reveal Functional Enhancer RNAs :ATTACH:
:PROPERTIES:
:ID:       d2b368ba-de24-48ff-a497-6012a72fd306
:END:

[[attachment:_20220408_195834screenshot.png]]

** GRO-Seq Overview :ATTACH:
:PROPERTIES:
:ID:       08136bc2-5fce-4dbb-bdb3-14793c5261d3
:END:


#+attr_latex: :height 0.18\linewidth
[[attachment:_20220411_121046screenshot.png]]

#+attr_latex: :height 0.18\linewidth
#+caption: [[https://www.illumina.com/science/sequencing-method-explorer/kits-and-arrays/5--gro-seq.html][Illumina Sequencing Methods]]
[[attachment:_20220411_121114screenshot.png]]


** GRO-Seq pros and cons
*** Pros:

- Maps nascent capped 5' RNA sequence at any given time
- Determines activity of transcription sites
- No prior knowledge of transcription sites needed

*** Cons:

- Limited to cell cultures and other artificial systems due to incubation with
  labelled nucleotides

** Kim et. al 2018 Summary :ATTACH:
:PROPERTIES:
:ID:       38cec5e3-c50e-4d76-9126-cf4883249f97
:END:

#+caption[Short caption]: [cite:@Kim_2018]
[[attachment:_20220411_101902screenshot.png]]


** Kim et. al 2018 Summary :ATTACH:
:PROPERTIES:
:ID:       754a799c-2e12-4613-9543-8d4311f83462
:END:

#+attr_latex: :height 0.25\linewidth
[[attachment:newplot.png]]

** Kim et. al 2018 Summary :ATTACH:
:PROPERTIES:
:ID:       fab0e52a-4588-4f5b-a045-debb7f415eeb
:END:

#+caption[Short caption]: [cite:@Kim_2018]
[[attachment:_20220411_102139screenshot.png]]



** Kim et. al 2018 Summary :ATTACH:
:PROPERTIES:
:ID:       28c3910e-fa74-4824-a617-3598712149e4
:END:

#+caption[Short caption]: [cite:@Kim_2018]
[[attachment:_20220411_102205screenshot.png]]

** Reproduction with Parallel IMR Dataset

- Wrote workflow using snakemake
- Goal was to reproduce GM results
- Achieved *80%* of predicted eRNAs due to difficulty with nascent transcript
  identification

** Hypothesis

- Does the standardization of secondary analysis and use of transcription start
  sites for calling enhancer RNAs improve the accuracy full transcript
  identification?
- Using the streamlined process of transcript identification can new dynamics
  and classes of eRNAs can be indentified from massively parrallel processing of
  publicly accessible nascent trancript assay across cell lines?
* Aims
* Aim 1: Create a best practice secondary analysis pipeline for nascent transcripts
** Standardizing Snakemake Workflow

- January 2020
- Template
- Universal Commands
- Testing
- CI/CD
- Wrappers

** nf-core Paper :ATTACH:
:PROPERTIES:
:ID:       4419713d-e23c-403b-b4e2-0c30201bfbac
:END:

[[attachment:_20220411_115734screenshot.png]]

** Main concepts of nf-core :ATTACH:
:PROPERTIES:
:ID:       b9be7b67-b57f-4f58-8cda-36455fb83c53
:END:

#+caption: [cite:@Ewels_2020]
#+attr_latex: :height 0.6\linewidth
[[attachment:_20220411_115958screenshot.png]]




** nf-core Getting started

#+attr_latex: :height 0.7\linewidth
#+begin_src bash
# Install nextflow
curl -s https://get.nextflow.io | bash
mv nextflow ~/bin/

# Launch the Nascent pipeline
nextflow run nf-core/nascent \
    --input samplesheet.csv \
    --genome GRCh38 \
    -profile docker
#+end_src

** Inheriting nf-core Nascent

- Breaking our analysis up into smaller pieces
- nf-core portion includes Quality Checks, alignment, graph generation, transcript
  identification, and transcript quantification
- Downstream analysis is then a seperate nextflow workflow
- Data engineering/Data Science split

** Primary-Secondary-Tertiary Analysis :ATTACH:
:PROPERTIES:
:ID:       31b1315d-d636-4cd7-928c-db5f94855d6f
:END:


[[attachment:_20220411_104006screenshot.png]]

** Nascent Goals

- Benchmark aligners to find best practices(Lots of opinions, no hard numbers)
- Handle alignment, QC, Genome graph generation, and naive transcript
  identification

* Aim 2: Take advantage of New Developments to improve eRNA annotation
** New developments

- CHM13 Released
- PINTS and transcriptional regulatory elements (TREs) matrix

** CHM13 :ATTACH:
:PROPERTIES:
:ID:       8087b00d-bcee-415d-99c3-34afae64e91a
:END:

#+caption: [cite:@Timp_2022]
[[attachment:_20220411_101653screenshot.png]]

** CHM13 :ATTACH:
:PROPERTIES:
:ID:       8269b2b8-4cb6-4bf3-a0c1-3604fdd4d423
:END:

#+caption: [cite:@Timp_2022]
[[attachment:_20220411_101552screenshot.png]]



** PINTS - different patterns of signals captured by TSS and NT Assays :ATTACH:
:PROPERTIES:
:ID:       483e1795-ac17-4e6b-a14b-37222b74a24d
:END:

#+attr_latex: :height 0.6\linewidth
#+caption: [cite:@Yu_2022]
[[attachment:_20220411_103318screenshot.png]]


** PINTS - NT vs TSS assays :ATTACH:
:PROPERTIES:
:ID:       cb525ffe-5925-48af-a434-cff675b835be
:END:

#+caption: [cite:@Yu_2022]
#+attr_latex: :height 0.35\linewidth
[[attachment:_20220408_112049screenshot.png]]

** PINTS :ATTACH:
:PROPERTIES:
:ID:       47b91df4-1791-4121-9c6c-ee1c403115b1
:END:

#+caption: [cite:@Yu_2022]
[[attachment:_20220411_103434screenshot.png]]


** PINTS

- Can we use PINTS for Nascent Transcript Assays?
- Can we swap naive method of selecting for Histone modications with PINTS
  indentified transcriptional regulatory elements (TREs)?
- Can we indetify full length transcripts from the Nascent Transcript assays?

* Aim 3 Compare eRNA dynamics between cell lines
** IMR and GM

- GM12878 - lymphoblastoid(immune response) cell line produced from the blood of
  a female donor (ENCODE Tier 1)
- IMR90 - fibroblasts(connective tissue) isolated from the normal lung
  tissue(ENCODE Tier 2.5)

** IMR and GM :ATTACH:
:PROPERTIES:
:ID:       7569a87c-da73-4704-a6b9-86b1ca1150d4
:END:

[[attachment:_20220411_105101screenshot.png]]
** Cell-Type Specific eRNAs

| Common inducible genes:                               | 2512 |
| Common inducible genes with cell-type specific eRNAs: | 2053 |

- 81.7%  Common inducible genes had *cell-type* specific eRNAs
- While they may have a common gene expression each cell line had their own
  unique way of solving the need.