#+title: Unlocking the Power of Genomic Analysis in Julia
#+author: Edmund Miller
#+date: July 26th, 2023

#+language: en
#+startup: content
#+property: header-args:conf-data-toml :eval no :exports code
#+property: header-args:julia :eval no :exports code

#+latex_class_options: [aspectratio=32]
#+beamer_theme: chameleon
#+latex_header: \usepackage{scrextend}
#+latex_header: \usepackage{xcolor}
#+beamer_theme: [progressbar=foot]metropolis
#+options: num:nil
#+options: toc:nil
#+startup: inlineimages
#+startup: beamer
#+LaTeX_CLASS: beamer
#+LaTeX_CLASS_OPTIONS: [bigger]
#+beamer_frame_level: 2
#+latex_header: \usepackage[scale=0.6]{sourcecodepro}
#+options: H:2
# SPC m e l O
#+chameleon_theme: doom-one-light
#+latex_engraved_theme: doom-one-light
#+latex_header: \usepackage[inkscapelatex=false,inkscapearea=page]{svg}
#+latex_header: \titlegraphic{\hfill\includesvg[height=2cm]{figures/logo.svg}}
#+latex_header: \usepackage[scale=0.88]{sourcecodepro}
#+latex_header: \DeclareUnicodeCharacter{2218}{\ensuremath{\circ}}
#+latex_header: \newcommand{\codefont}{\footnotesize\fontseries{mb}\selectfont}
# #+latex_header: \usepackage{appendixnumberbeamer}
#+latex_header_extra: \definecolor{EfD}{HTML}{fafafa}
# FIXME What does this do?
#+latex_header_extra: \setbeamertemplate{headline}{Hello}

# https://pretalx.com/juliacon2023/me/submissions/AJJRS3/

* PROJ Tracking :noexport:
- [ ] Fix backticks being quotes
- [ ] Reattach R call graph

* Abstract :noexport:

Learn how Julia, a high-performance programming language, can be used to analyze
genomic data. Discussion of libraries, specific challenges and opportunities,
past examples, and future possibilities of using Julia in genomic data analysis.

* Description :noexport:

Genomic data is becoming an increasingly valuable resource in the study of
biology and medicine, as it allows for a deeper understanding of the underlying
mechanisms of diseases and the development of more effective therapies. However,
the sheer volume and complexity of genomic data can make it challenging to
analyze. Julia, a high-performance programming language, has emerged as a
powerful tool for genomic data analysis. In this talk, we will explore the use
of Julia for genomic data analysis, including the various libraries and packages
available, such as IntervalTrees and GenomicFeatures. We will also discuss some
of the specific challenges and opportunities that arise when analyzing genomic
data, such as dealing with large-scale data and integrating multiple data types.
We will also show some examples of how Julia has been used in the past to
analyze genomic data and what the future holds for this field. This talk will be
beneficial for biologists, bioinformaticians, and data scientists interested in
the application of Julia to genomic data analysis.

Expected Outcomes:

- Understanding of the power and capabilities of Julia for genomic data analysis
- Knowledge of the available libraries and packages for genomic data analysis in
  Julia
- Insights into the challenges and opportunities of using Julia for genomic data
  analysis
- Familiarity with examples of how Julia has been used in the past for genomic
  data analysis
- Ideas for potential future applications of Julia in genomic data analysis.

* Overview
** TODO ~whoami~

- Phd Candidate @ University of Texas at Dallas
- nf-core maintainer
** Why Julia and Genomics?
** Why are biologists working in Genomics interested?

- "Why We Created Julia"
  - As easy for statistics as R
  - With mathematical notation like Matlab
  - As usable for general programming as Python
  - As natural for string processing as Perl
  - As good at gluing programs together as the shell
  - As fast as C

- Reproducibility
# TODO cite: https://julialang.org/blog/2012/02/why-we-created-julia/
** Why should Julia enthusiasts interested in Genomics?

- Rapid lowering cost of sequencing cost
- Gene Therapy
- Interest in keeping costs low through more efficient computation
    - Cost of Whole Genome Sequencing: $100
    - Cost of computational Analysis: $25

# TODO cite: Sarek 3.0 paper
*** :B_note:
:PROPERTIES:
:BEAMER_env: note
:END:

- Smaller computers needed that run in less time

** Overview

- Julia Features for Analysis
- Ecosystem
- Julia in Workflows

* Julia Features for Analysis
** Juliaup

# TODO Make this bigger
#+begin_src bash
curl -fsSL https://install.julialang.org | sh
#+end_src

- Cross-platform installer
- Install specific Julia versions
- Stay up to date with release channels(Stable, LTS)

** Package management - Pkg.jl
#+begin_src julia
# ]
(@v1.8) pkg>
(@v1.8) pkg> add Example
Resolving package versions...
Installed Example ─ v0.5.3
    Updating `~/.julia/environments/v1.8/Project.toml`
[7876af07] + Example v0.5.3
    Updating `~/.julia/environments/v1.8/Manifest.toml`
[7876af07] + Example v0.5.3
julia> import Example

julia> Example.hello("friend")
"Hello, friend"
#+end_src
** Pkg.jl Environments

#+begin_src julia
(@v1.8) pkg> activate tutorial
[ Info: activating new environment at `~/tutorial/Project.toml`.
(tutorial) pkg>
(tutorial) pkg> status
    Status `~/tutorial/Project.toml`
(empty environment)
(tutorial) pkg> add Example JSON
...

(tutorial) pkg> status
    Status `~/tutorial/Project.toml`
[7876af07] Example v0.5.3
[682c06a0] JSON v0.21.3
#+end_src

** Want to hack on a project?

#+begin_src julia
pkg> develop --local Muon
#+end_src

There's a full clone at ~dev/Muon~!

** TODO Easy Package Creation

[[https://github.com/JuliaCI/PkgTemplates.jl][PkgTemplates.jl]]

- CI template
- When you jump into a unfamiliar repo you know where everything is

** TODO REPL :noexport:

- Allows you to "talk" to the computer
- Jupyter without the pain

** TODO VS Code? :noexport:
- Remote files
** DataToolkit
# TODO Insert Logo
** DataToolkit - Example declarative data set

\small
#+begin_src conf-data-toml
[[iris]]
uuid = "3f3d7714-22aa-4555-a950-78f43b74b81c"
description = "Fisher's famous Iris flower measurements"

    [[iris.storage]]
    driver = "web"
    checksum = "crc32c:d5c06b86"
    url = "https://raw.githubusercontent.com/scikit-learn/scikit-learn/1.0/sklearn/datasets/data/iris.csv"

    [[iris.loader]]
    driver = "csv"
    args.header = ["sepal_length", "sepal_width", "petal_length", "petal_width", "species_class"]
    args.skipto = 2
#+end_src

** DataToolkit - Features in this example

+ A named dataset @@latex:\hfill@@
  src_conf-data-toml{[[iris]]}
+ Which can be uniquely identified @@latex:\hfill@@
  src_conf-data-toml{uuid = "..."}
+ With metadata @@latex:\hfill@@
  src_conf-data-toml{description = "..."}
+ Named storage/loader backends @@latex:\hfill@@
  src_conf-data-toml{driver = "web"}
+ Content verification @@latex:\hfill@@
  src_conf-data-toml{checksum = "crc32c:d5c06b86"}
+ Storage/loader arguments @@latex:\hfill@@
  src_conf-data-toml{url = "..."}, src_conf-data-toml{args.skipto = 2}

** DataToolkit - Using a dataset in computation

#+begin_src julia
julia> using DataToolkit, DataFrames

julia> sum(d"iris".sepal_length)
876.5
#+end_src

#+beamer: \pause

#+begin_src julia
julia> mean(d"iris::Matrix", dims=1)
1×5 Matrix{Float64}:
 5.84333  3.05733  3.758  1.19933  1.0
#+end_src

#+beamer: \pause

More than just string matching for types:

#+begin_src julia
julia> mean(d"iris::Array{T<:Any, 2}", dims=1)
1×5 Matrix{Float64}:
 5.84333  3.05733  3.758  1.19933  1.0
#+end_src

** TODO DataToolkit - Want to learn more?
# Link to Teco's presentation

Join Teco Friday @ TODO

* Ecosystem
:LOGBOOK:
CLOCK: [2023-07-15 Sat 21:30]--[2023-07-15 Sat 22:27] =>  0:57
CLOCK: [2023-07-15 Sat 14:17]--[2023-07-15 Sat 14:47] =>  0:30
:END:

** Package comparisons - General Utilities
\scriptsize
| Purpose    | Python        | R       | Julia                                  |
|------------+---------------+---------+----------------------------------------|
| /          | <>            | <>      | <>                                     |
| Plotting   | Matplotlib    | ggplot2 | [[https://github.com/JuliaPy/PyPlot.jl][PyPlot.jl]] / [[https://github.com/MakieOrg/Makie.jl][Makie.jl]] / [[https://gadflyjl.org/stable/][Gadfly.jl]] |
| Dataframes | Pandas/Polars | [[https://tibble.tidyverse.org/][tibble]]  | DataFrames.jl                          |

** Package comparisons - Biological File Formats

\scriptsize
| Purpose       | Python    | R  | Julia                                    |
|---------------+-----------+----+------------------------------------------|
| /             | <>        | <> | <>                                       |
| Sam/Bam files | [[https://biopython.org/wiki/SeqIO][Bio.SeqIO]] |    | [[https://docs.juliahub.com/XAM/4JnDO/0.3.1/][XAM.jl]]                                   |
| Fastq files   |           |    | [[https://github.com/BioJulia/FASTX.jl][FASTX.jl]]                                 |
| Variants/vcf  |           |    | [[https://github.com/BioJulia/GeneticVariation.jl][GeneticVariation.jl]] / [[https://github.com/rasmushenningsson/VariantCallFormat.jl][VariantCallFormat.jl]] |
|               |           |    | [[https://biojulia.dev/Phylogenies.jl/stable/][Phylogenies.jl]]                           |
|               | [[https://biopython.org/wiki/The_Biopython_Structural_Bioinformatics_FAQ][Bio.PDB]]   |    | [[https://biojulia.dev/BioStructures.jl/stable/][BioStructures.jl]]                         |
|               |           |    | [[https://github.com/BioJulia/GFF3.jl][GFF3.jl]]                                  |

*** :B_note:
:PROPERTIES:
:BEAMER_env: note
:END:

- Gadfly - if you prefer Grammer of graphics

https://www.bioconductor.org/packages/stats/

** Package comparisons - Genomic Analysis

\tiny
| Purpose             | Python                    | R              | Julia                          |
|---------------------+---------------------------+----------------+--------------------------------|
| /                   | <>                        | <>             | <>                             |
|                     | [[https://github.com/pyranges/pyranges][pyranges]] / [[https://daler.github.io/pybedtools/][pybedtools]]     | [[https://bioconductor.org/packages/release/bioc/html/rtracklayer.html][rtracklayer]]    | [[https://docs.juliahub.com/GenomicFeatures/kSGNI/3.0.0/][GenomicFeatures.jl]]             |
|                     |                           | [[https://bioconductor.org/packages/release/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesIntroduction.html][GenomicsRanges]] | [[https://biojulia.dev/GenomicFeatures.jl/stable/man/intervals/][Intervals]] |
|                     |                           | [[https://github.com/drostlab/metablastr][metablastr]]     | [[https://docs.juliahub.com/BioTools/wwbVn/1.1.0/blast/][BioTools.jl]]                    |
|                     | [[https://biopython.org/wiki/SeqIO][Bio.SeqIO]]                 |                | [[https://biojulia.dev/BioSequences.jl/stable/transforms/][BioSequences.jl]]                |
| Data Retrieval      | [[https://github.com/pyranges/pyranges_db][pyranges_\db]] / [[https://github.com/sebriois/biomart][biomart]] (api) | [[https://github.com/ropensci/biomartr][biomartr]]       | [[https://docs.juliahub.com/BioServices/nOcmO/0.4.1/man/eutils/][BioServices.jl]] / [[https://github.com/BioJulia/BioFetch.jl][BioFetch.jl]]   |
| Genomic Annotations |                           |                | [[https://docs.juliahub.com/GenomicAnnotations/ckOyU/0.3.2/][GenomicAnnotations.jl]]          |
| Population Genetics | [[https://biopython.org/wiki/PopGen][Bio.PopGen]]                |                | [[https://github.com/BioJulia/PopGen.jl][PopGen.jl]]                      |

** What about when you can't replace popular packages?

#+beamer: \pause
- Working in Python & R is like buying a _house in DFW_
#+beamer: \pause
  - Downtown, constant re-development(pip, poetry, hatch, piptools, conda)
#+beamer: \pause
  - Compared to the suburbs where you need a car

#+beamer: \pause
- Urban Sprawl of Python & R packages
  - DESeq2/edgeR/seurat/scanpy
  - ggplot2


*** :B_note:
:PROPERTIES:
:BEAMER_env: note
:END:

- Gotta give my PI credit for this one
- You're constantly getting new restaurants, there's plenty of public
  transportation
- But at least in the suburbs your favorite restaurant isn't getting torn down, hopefully there's less traffic.

** What about when you can't replace popular packages?

Options:
- [[https://github.com/JuliaInterop][JuliaInterop · GitHub]]
- [[https://juliainterop.github.io/RCall.jl/stable/gettingstarted/][RCall.jl]]
# FIXME PythonCall
- PythonCall.jl

- Calling commandline tools from Julia

*** :B_note:
:PROPERTIES:
:BEAMER_env: note
:END:

- Also libraries for C/C++, Matlab and GNU Octave, Java, Fortran
- There is a Rust crate but it doesn't seem very active. Probably because Julia
came out in 2009, and Rust in 2015, so why would you reach for Rust if you're
going to write Julia anyways.

There is a Rust birds of a feather

** TODO JuliaInterop - RCall :ATTACH:
:PROPERTIES:
:ID:       7f2a99d1-01a5-4432-8fc9-d8d3caf043c8
:END:

#+begin_src julia
julia> using RCall
# type `$`
R> install.packages("ggplot2")
R> library(ggplot2)
R> data(diamonds)
R> ggplot(diamonds, aes(x=carat, y=price)) \
    + geom_point()
#+end_src

# TODO Reattach
#+attr_latex: :height 0.2\linewidth

** TODO JuliaInterop - RCall importing packages

** TODO PythonCall
#+begin_src python
import scanpy as sc


def preprocessing(adata):
    # Perform preprocessing of a anndata object
    sc.pp.filter_cells(adata, min_genes=200)
    sc.pp.filter_genes(adata, min_cells=3)

    # Normalization and scaling:
    sc.pp.normalize_total(adata, target_sum=1e4)
    sc.pp.log1p(adata)

    # Identify highly-variable genes
    sc.pp.highly_variable_genes(
        adata, min_mean=0.0125, max_mean=3, min_disp=0.5, subset=True
    )
    sc.pp.scale(adata, zero_center=True, max_value=3)
    x = adata.X
    data = tf.data.Dataset.from_tensor_slices((x, x))
    return data, x
#+end_src
# TODO Make left and right
#+begin_src julia
using PythonCall

sc = pyimport("scanpy")

function preprocessing(adata)
    sc.pp.filter_cells(adata, min_genes=200)
    sc.pp.filter_genes(adata, min_cells=3)

    # Normalization and scaling:
    sc.pp.normalize_total(adata, target_sum=1e4)
    sc.pp.log1p(adata)

    # Identify highly-variable genes
    sc.pp.highly_variable_genes(adata, min_mean=0.0125, max_mean=3, min_disp=0.5, subset=true)
    sc.pp.scale(adata, zero_center=true, max_value=3)
    x = adata.X
    # We don't need Tensorflow because Julia is fast enough I think?
    # data = tf.data.Dataset.from_tensor_slices((x, x))
    x = pyconvert(Array{Float32}, x)

    return [x, x], x
end
#+end_src
** PythonCall and Pycall are different

- Doesn't have to support as much legacy
  - PythonCall supports Julia 1.6.1+ and Python 3.7+
  - PyCall supports Julia 0.7+ and Python 2.7+.
- Uses CondaPkg by default
- You can use them both at the same time if you needed to for some reason

** Managing conda envs in Julia

#+begin_src julia
using Conda, RCall

Conda.add("bioconductor-deseq2", channel="bioconda", :rnaseq)
#+end_src

# ~ENV["CONDA_JL_USE_MINIFORGE"] = "1"~
# julia> using Conda
# julia> Conda.add("mamba")
# julia> ENV["CONDA_JL_CONDA_EXE"] = joinpath(Conda.ROOTENV, "bin", "mamba")
# pkg> build Conda

** TODO CondaPkg.jl
** Calling commandline tools from Julia

#+begin_src julia
julia> mycommand = `echo hello`
`echo hello`

julia> typeof(mycommand)
Cmd

julia> run(mycommand);
hello
#+end_src

# TODO Cite
[[https://docs.julialang.org/en/v1/manual/running-external-programs/][Docs on Running External Programs]]
** Calling commandline tools from Julia
# TODO Convert this to a bioinformatics example?

\small
#+begin_src julia
julia> files = ["/etc/passwd","/Volumes/External HD/data.csv"]
2-element Vector{String}:
"/etc/passwd"
"/Volumes/External HD/data.csv"

julia> `grep foo $files`
`grep foo /etc/passwd '/Volumes/External HD/data.csv'`
#+end_src

** TODO Plotting - IGV with Plotly.js :noexport:

#+begin_src julia
using Dash, DashBio

app = dash()

app.layout = dashbio_igv(
    id="genome-igv",
    genome="ce11"
)

run_server(app, "0.0.0.0", debug=true)

#+end_src
** TODO Plotting - BedGraphs - Example :noexport:

\small
#+begin_src bedgraph
track type=bedGraph name="BedGraph Format" description="BedGraph format" priority=20
chr19 59302000 59302300 -1.0
chr19 59302300 59302600 -0.75
chr19 59302600 59302900 -0.50
chr19 59302900 59303200 -0.25
chr19 59303200 59303500 0.0
chr19 59303500 59303800 0.25
chr19 59303800 59304100 0.50
chr19 59304100 59304400 0.75
#+end_src
** Plotting - BedGraphs :noexport:

\small
#+begin_src julia :using DataFrames FileIO
using FileIO, BedgraphFiles, DataTables, IndexedTables, Gadfly

# Load into a DataTable
dt = DataTable(load("data.bedgraph"))

# Load into an IndexedTable
it = IndexedTable(load("data.bedgraph"))
# Plot directly with Gadfly
plot(load("data.bedgraph"), xmin=:leftposition, xmax=:rightposition, y=:value, Geom.bar)

load("data.bedgraph") |> @filter(_.chrom == "chr19") |> save("data-chr19.bedgraph")
#+end_src

# FIXME gadly and indexedtable are broken

** TODO Mention Michael's new project
* Julia in Workflows
** Running Julia in Snakemake

#+begin_src snakemake
from snakemake.remote import AUTO
iris = "https://raw.githubusercontent.com/scikit-learn/scikit-learn/1.0/sklearn/datasets/data/iris.csv"
rule calling_script:
    input:
        AUTO.remote(iris)
    output:
        "results/out.csv",
    container: "docker://julia"
    script:
        "bin/smk_script.jl"
#+end_src

\small
#+begin_quote
In the Julia script, a snakemake object is available, which can be accessed
similar to the Python case, with the only difference that you have to index from
1 instead of 0.
#+end_quote

** Running Julia in Snakemake - Inside the Julia script
# FIXME make this a subtitle

#+begin_src julia
import Pkg; Pkg.add(["CSV", "DataFrames"])

using CSV, DataFrames

df = DataFrame(CSV.File(snakemake.input[1], footerskip=50))
names(df)
CSV.write(snakemake.output[1], df)

do_something(snakemake.input[1], snakemake.output[2], snakemake.threads, snakemake.config["myparam"])
#+end_src

** TODO Handling package installs :noexport:
https://github.com/snakemake/snakemake/issues/2215

** Running Julia in Nextflow - Installing Packages to Julia Depot

[[https://apeltzer.github.io/post/03-julia-lang-nextflow/][Julia Lang, Docker & Nextflow | Personal Homepage of Alex Peltzer]]

#+begin_src nextflow
// nextflow.config
env {
    JULIA_DEPOT_PATH = "/usr/local/share/julia"
}
#+end_src


** Running Julia in Nextflow - The Nextflow script

\small
#+begin_src nextflow
process cli {
    container 'julia'

    input:
    path csv_file

    output:
    stdout

    """
    julia hello.jl $csv_file
    """
}

process shebang {
    container 'julia'
    beforeScript "julia -e 'using Pkg; Pkg.activate("."); Pkg.add(["HTTP", "DataFrames"]); Pkg.precompile();'"

    input:
    path csv_file

    output:
    path "out.csv"

    """
    #!/usr/bin/env -S julia --startup-file=no

    using CSV, DataFrames

    df = DataFrame(CSV.File($csv_file, footerskip=50))
    names(df)
    CSV.write("out.csv", df)
    """
}

workflow {
    cli(file('./test.csv'))
    shebang(file('./test.csv'))
}
#+end_src

** Running Julia in Nextflow - The Julia script
#+begin_src julia
#!/usr/bin/env -S julia --color=yes --startup-file=no

println(PROGRAM_FILE);
abspath(PROGRAM_FILE) == @__FILE__

@show ARGS

for x in ARGS
    println(x)
end
#+end_src

- Move it to the ~bin/~ folder of the pipeline, and make it executable (~chmod +x bin\*.jl~)


** :B_note:
:PROPERTIES:
:BEAMER_env: note
:END:

- The ~--project=@.~ is the default
- But the way Nextflow works that doesn't get picked up
** TODO Handling package installs :noexport:

https://github.com/JuliaContainerization/SimpleContainerGenerator.jl
* TODO Analysis Example

#+begin_src julia
# overlap of H3K27ac and P63 peaks identifies enhancer regions where p63 binds

using Downloads

if !isfile(raw"H3K27ac.consensus_peaks.bed")
    Downloads.download("https://utdallas.box.com/shared/static/y4glk8y8chjq5fuv6iowe6bjz1vovkr3.bed", "H3K27ac.consensus_peaks.bed")
    Downloads.download("https://utdallas.box.com/shared/static/y4glk8y8chjq5fuv6iowe6bjz1vovkr3.bed", "p63_4A4.consensus_peaks.bed")
end

using GenomicFeatures
using BED

# Create an interval collection in memory.
h3k27ac_icol = open(BED.Reader, "H3K27ac.consensus_peaks.bed") do reader
    IntervalCollection(reader)
end

p63_icol = open(BED.Reader, "p63_4A4.consensus_peaks.bed") do reader
    IntervalCollection(reader)
end

overlap_icol = eachoverlap(h3k27ac_icol, p63_icol)
typeof(p63_icol)
first(overlap_icol)
overlaps = collect(overlap_icol)
overlaps
typeof(overlaps[1][1])
BED.Record(p63_icol)

writer = BED.Writer(output)
expected_entries = BED.Record[]
for interval in open(BED.Reader, filename)
    write(writer, interval)
    push!(expected_entries, interval)
end
#+end_src

* Conclusion
** Where is Julia lacking?

- Creating binaries/CLIs
- But what about Rust?
  - Rust for tools
  - Julia for analysis
** Resources

- [[https://github.com/BioJulia/BioTutorials][GitHub - BioJulia/BioTutorials: Tutorial Notebooks of BioJulia]]
- New Documenter.jl Docs!
- TODO Slides Link
- [[https:/link.edmundmiller.dev][link.edmundmiller.dev]]

* Ideas :noexport:
** Other ideas to mimic
*** Look at genomicranges example workflow.
https://bioconductor.org/packages/release/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesIntroduction.html
*** Crazy hot Tommy's blogs
*** Other genomic R package demos?
** TODO [[https://github.com/compbiocore/VariantVisualization.jl][GitHub - compbiocore/VariantVisualization.jl: Julia package powering VIVA, ou...]]
