#+title: Exploring the State of Machine Learning for Biological Data
#+author: Edmund Miller
#+language: en
#+date: July 26th, 2023
#+exclude_tags: noexport
#+options: num:nil
#+options: toc:nil
#+startup: inlineimages
#+startup: beamer
#+LaTeX_CLASS: beamer
#+LaTeX_CLASS_OPTIONS: [bigger]
#+beamer_frame_level: 2
# SPC m e l O

# https://pretalx.com/juliacon2023/me/submissions/CSG8NU/

* Abstract :noexport:

Exploring the use of Julia, in analyzing biological data. Discussion of libraries and packages, challenges and opportunities of using machine learning on biological data, and examples of past and future applications.

* Description :noexport:

This talk, "Exploring the State of Machine Learning for Biological Data in Julia," will delve into the use of the high-performance programming language, Julia, in analyzing biological data. We will discuss various libraries and packages available in Julia, such as BioJulia and Flux.jl, and the benefits of using Julia for machine learning in the field of biology. Additionally, the challenges and opportunities that arise when using machine learning techniques on biological data, such as dealing with high-dimensional and heterogeneous data, will be addressed. The talk will also include examples of how machine learning has been applied to biological data in the past and what the future holds for this field.

* Exploring the State of Machine Learning for Biological Data

#+begin_quote
Be the PR you want to see in the repo
#+end_quote

* Exploring the State of Machine Learning for Biological Data

#+begin_quote
Be the +PR+ Talk you want to see +in the repo+ at Juliacon
#+end_quote



* About me

- Phd Candidate @ University of Texas at Dallas
- nf-core maintainer

#+beamer: \pause

- Been Machine Learning curious since around 2017
- Never took Linear Algebra

#+beamer: \pause

- I had some questions
  - What are the trade offs of all these packages?
  - How do we load biological data? (Personal interest in genomics)

* Goal of Biologists
# scope

- We're not trying to create novel models
- We're trying to apply these models in novel ways
- Then make biological inferences


* Old overview

- Ability to call Python and R packages
- What are all of these different ML Packages?
- Loading biological file formats

* Then I had a thought...

#+beamer: \pause
- What if

* Ability to call Python and R packages
https://github.com/instadeepai/nucleotide-transformer

* What are all of these Packages?

- Flux.jl
- SciML
- MLJ
- fast.ai
- MLJFlux.jl

* Loading biological file formats

- FASTA
- BAM
- BED

* Somewhat bleak, but python and R aren't much better
* Reproduce https://github.com/mlf-core/lcep?
https://juliapackages.com/p/xgboost
* Maybe just attempt to reproduce all of mlf-core examples?
Start with lcep
- Do SC autoencoder
https://github.com/mlf-core/liver-ct-segmentation

* Calling Selene
https://github.com/FunctionLab/selene/blob/master/tutorials/quickstart_training/quickstart_training.ipynb
https://github.com/FunctionLab/selene/blob/master/tutorials/getting_started_with_selene/getting_started_with_selene.ipynb
* Hooking up to CUDA
https://fluxml.ai/Flux.jl/stable/gpu/
* Using it in collab
https://colab.research.google.com/github/ageron/julia_notebooks/blob/master/Julia_Colab_Notebook_Template.ipynb
* Flux
https://fluxml.ai/Flux.jl/stable/gpu/
https://www.freecodecamp.org/news/deep-learning-with-julia/
* Conclusion

- I need to read a few books on linear algebra


* Exploring the State of Machine Learning for Biological Data
** Things I want to cover
- Loading file biological file formats

- Goal of Biologists using machine learning
  - We're not trying to create novel model
  - We're trying to apply these models in novel ways
  - Then make biological inferences

** Reproduction of mlf-core examples

- [[https://github.com/mlf-core/sc-autoencoder][An autoencoder for single cell data.]]
- [[https://github.com/mlf-core/lcep][Classifying cancerous liver samples from gene expression data.]]
- [[https://github.com/mlf-core/liver-ct-segmentation][Liver-tumor segmentation of computed tomography scans using a U-Net model.]]
