% Created 2021-10-12 Tue 17:50
% Intended LaTeX compiler: pdflatex
\documentclass[bigger]{beamer}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{graphicx}
\usepackage{grffile}
\usepackage{longtable}
\usepackage{wrapfig}
\usepackage{rotating}
\usepackage[normalem]{ulem}
\usepackage{amsmath}
\usepackage{textcomp}
\usepackage{amssymb}
\usepackage{capt-of}
\usepackage{hyperref}
\usetheme[progressbar=foot]{metropolis}
\author{Edmund Miller}
\date{2021-10-13 Wed}
\title{Research Update}
\hypersetup{
 pdfauthor={Edmund Miller},
 pdftitle={Research Update},
 pdfkeywords={},
 pdfsubject={Internship with Element Biosciences},
 pdfcreator={Emacs 28.0.50 (Org mode 9.4.4)}, 
 pdflang={English}}
\begin{document}

\maketitle

\section{Internship with Element Biosciences}
\label{sec:orgb717863}

\begin{frame}[label={sec:orga69606c}]{Element Biosciences}
\begin{itemize}
\item Startup creating a new DNA sequencing platform
\item Worked with the Bioinfomatics group
\item Met Bryan Lajoie through nf-core
\end{itemize}
\end{frame}

\begin{frame}[label={sec:org0dc6ff7}]{Overview}
\begin{itemize}
\item ERCC Analysis
\item COVID Assay analysis
\item Secondary Analysis Infrastructure
\end{itemize}
\end{frame}

\section{ERCC Analysis - External RNA Controls Consortium}
\label{sec:orgb301009}
\begin{frame}[label={sec:org61ffe79}]{ERCC Analysis - External RNA Controls Consortium}
\begin{itemize}
\item Evaluation of multiple performance characteristics
\begin{itemize}
\item Linear performance of individual controls
\item Signal response within dynamic range pools of controls
\item Ratio detection between pairs of dynamic range pools.
\end{itemize}
\end{itemize}

\begin{columns}
\begin{column}{0.35\columnwidth}
\begin{center}
\includegraphics[width=\textwidth]{/home/emiller/sync/org/roam/data/ba/aefd37-898b-423a-a702-7d767f64f391/_20211004_163409screenshot.png}
\end{center}
\end{column}

\begin{column}{0.5\columnwidth}
\begin{center}
\includegraphics[width=\textwidth]{/home/emiller/sync/org/roam/data/97/d9a007-e078-45c0-9b8e-5659ed9a438b/_20211004_163631screenshot.png}
\end{center}
\end{column}
\end{columns}
\end{frame}



\begin{frame}[label={sec:org3938afa}]{ERCC Analysis}
\begin{itemize}
\item Allows for estimation of Lab to Lab (instrument to instrument) variation.
\item Used \href{https://www.bioconductor.org/packages/release/bioc/html/erccdashboard.html}{erccdashboard} to create a standardized analysis.
\end{itemize}
\end{frame}

\section{Amplicon Analysis for Covid assay}
\label{sec:orgc69a70b}
\begin{frame}[label={sec:org89ee4af}]{Amplicon Analysis for Covid assay}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/94/536871-7213-4228-a9a4-3ac6d0fba1e8/_20211004_165109screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:org1a14953}]{nf-core/viralrecon}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/84/ad5978-ba58-463c-ba72-9747e7cbea22/_20211004_165305screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:org34e0721}]{nf-core/viralrecon}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/df/5ed11b-ac78-4cb4-b44f-44b65eea42ba/_20211004_165747screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:orgcdb020f}]{nf-core/viralrecon}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/e1/8c87a6-f9f4-4187-8d76-ee2bdfc6af0b/_20211004_170305screenshot.png}
\end{center}
\end{frame}


\begin{frame}[label={sec:org5f233a6}]{Analysis of Covid Variants}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/6d/9cf682-29d6-4ad5-b2d3-32c4115ba070/_20211004_170948screenshot.png}
\end{center}
\end{frame}

\section{Secondary Analysis Infrastructure}
\label{sec:orgb0e3f13}

\begin{frame}[label={sec:org48935f3}]{What is Secondary Analysis?}
Primary Analysis - Specific steps needed to transform images into base-calls and compute quality scores for those bases

Secondary Analysis – Alignment of these short sequencing reads onto a reference genome and variant calling

Tertiary Analysis – Interpreting the secondary analysis data (annotation, qc metrics, filtering, benchmarking)
\end{frame}

\begin{frame}[label={sec:org46ffab4}]{Types of Secondary Analysis}
\begin{itemize}
\item WGS
\begin{itemize}
\item Human, ecoli, phix, covid
\end{itemize}
\item WES
\begin{itemize}
\item Exome, panel, amplicon
\end{itemize}
\item Single Cell
\begin{itemize}
\item 10x scRNA-Seq, 10x spatial, 10x scATAC-Seq
\end{itemize}
\item RNA-Seq
\begin{itemize}
\item Bulk RNA
\end{itemize}
\item MetaGenomics
\begin{itemize}
\item Stool sample
\end{itemize}
\end{itemize}
\end{frame}

\begin{frame}[label={sec:orgfb2d51b}]{Whole Genome Sequencing}
\begin{itemize}
\item Allows us to compare with the “truth”

\item Genome in a Bottle
\begin{itemize}
\item Leveraging multiple instrument platforms to create truth datasets
\item Truth is available for HG001-HG005 with diverse genetic backgrounds
\end{itemize}

\item Allowed us to provide feedback to the rest of the teams
\begin{itemize}
\item Context Errors
\end{itemize}
\end{itemize}
\end{frame}

\begin{frame}[label={sec:org8c84131}]{Goals of the Secondary Analysis Infrastructure}
\begin{itemize}
\item Mimicking a Customer environment
\item Internal Data discoverability
\item \alert{Automation}
\end{itemize}
\end{frame}

\begin{frame}[label={sec:org4a8af58}]{Design Decisions}
\begin{columns}
\begin{column}{0.35\columnwidth}
\begin{center}
\includegraphics[width=\textwidth]{/home/emiller/sync/org/roam/data/4f/b43e6a-fd94-4123-94fb-9eb49d01bbef/_20211004_173812screenshot.png}
\end{center}


\begin{center}
\includegraphics[width=\textwidth]{/home/emiller/sync/org/roam/data/4f/b43e6a-fd94-4123-94fb-9eb49d01bbef/_20211004_173820screenshot.png}
\end{center}
\end{column}

\begin{column}{0.5\columnwidth}
\begin{center}
\includegraphics[width=\textwidth]{/home/emiller/sync/org/roam/data/8a/619033-8dba-48de-a2a1-3b3e563fa007/_20211005_103739screenshot.png}
\end{center}


\begin{center}
\includegraphics[width=\textwidth]{/home/emiller/sync/org/roam/data/8a/619033-8dba-48de-a2a1-3b3e563fa007/_20211004_174248screenshot.png}
\end{center}
\end{column}
\end{columns}
\end{frame}

\begin{frame}[label={sec:orgb04d7e5}]{Nextflow}
\begin{itemize}
\item Opensource
\begin{itemize}
\item Supported by Seqera
\end{itemize}

\item Platform independent
\begin{itemize}
\item Runs locally, Cloud, SLURM, hybrid
\end{itemize}

\item Reproducibility
\end{itemize}
\end{frame}

\begin{frame}[label={sec:orgd7ba3b8}]{nf-core}
\begin{itemize}
\item Common bioinformatics software modules make creating new workflows quickly

\item Curated set of best practice pipelines to avoid reinventing the wheel for
\alert{secondary analysis}.

\item Template to quickly start new pipelines
\end{itemize}
\end{frame}

\begin{frame}[label={sec:orgcf41bad}]{AWS Batch}
\begin{itemize}
\item Abstracts away the cluster provisioning
\item Spot Instances
\item Utilizing High Performance systems
\end{itemize}


\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/c8/7b16b3-cd2d-4d96-b18c-1da2204ec7fe/_20211005_095412screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:org463cb94}]{Nextflow Tower}
\begin{itemize}
\item Handling AWS batch environment
\item Monitoring, logging \& observability
\item Automation
\item Smoothing out Customer Experience
\end{itemize}
\end{frame}

\begin{frame}[label={sec:orgcf62980}]{Things learned from this Internship}
\begin{itemize}
\item Exposure to Cloud computing for bioinformatics
\item Improved my tertiary analysis skills
\item Exposure to primary analysis
\item Exposure to a greater variety of assays
\item Better understanding of job titles and roles that are out there
\item Skills Seymon looks for when hiring(in order):
\begin{itemize}
\item Ability to write production level code
\item Developing novel algorithms
\item Tertiary analysis skills
\end{itemize}
\end{itemize}
\end{frame}

\section{Notebook Template}
\label{sec:org5d577ea}
\begin{frame}[label={sec:orgf2ab28c}]{Notebook Template Goals}
\begin{itemize}
\item Creating a separation between secondary and tertiary analysis
\item Ingesting the expected results from secondary analysis
\item Environment is easily reproducible but flexible for moving quickly
\item Avoid being tied to one language
\item Data science instead of data engineering
\end{itemize}
\end{frame}

\begin{frame}[label={sec:orgee19fda},fragile]{Getting started}
 \begin{enumerate}
\item Go to \href{https://github.com/Functional-Genomics-Lab/notebook-template}{\alert{GitHub - Functional-Genomics-Lab/notebook-template}}
\item Click ``Use this Template''
\item \texttt{docker-compose up}
\item Copy the link to your local jupyter instance from the terminal and open it in
your browser.
\end{enumerate}
\end{frame}

\begin{frame}[label={sec:org9499d13}]{Selecting an Image}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/6a/f94bed-fcfd-4fac-89f0-c4207236ef91/_20211005_121802screenshot.png}
\end{center}
\end{frame}


\begin{frame}[label={sec:org08ab386}]{Quick Tour using GRO-Seq Analysis}
\begin{itemize}
\item Dockerfile
\item requirements.txt
\item Notebooks
\end{itemize}
\end{frame}

\begin{frame}[label={sec:org515f4d1}]{Inspiration from 10x}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/38/522a2a-53ca-48d9-b45d-58e5c630642c/_20211006_084828big_picture.png}
\end{center}
\end{frame}


\section{nf-core/nascent}
\label{sec:org817e2a6}

\begin{frame}[label={sec:orgeef3de3}]{nf-core/nascent}
\begin{itemize}
\item Taking over an old repo to avoid duplication of work and fragmenting community
\item Main purpose is going from FastQ to counts, nascent transcripts, and
bedGraph/bigWigs
\item The output files can be used in UCSC Genome Browser or in Notebooks
\end{itemize}
\end{frame}

\begin{frame}[label={sec:orga373cf2}]{Conversion Progress}
\begin{itemize}
\item Updated to the most recent nf-core template
\item Rebased our Commits on top of the old repo (To preserve v1.0 for any legacy
research)
\end{itemize}
\end{frame}

\begin{frame}[label={sec:orgb9f2d43}]{Things left TODO}
\begin{itemize}
\item Old nascent functionality added in a subworkflow
\item Add test data to nf-core test data
\item Refgenie nf-core infrastructure to use T2T-CHM13 reference
\end{itemize}
\end{frame}

\begin{frame}[label={sec:org2125b59}]{nf-core Hackathon}
\begin{itemize}
\item October 27th-29th 2021
\item Focus is going to be on converting pipelines to DSL2
\item \href{https://nf-co.re/events/2021/hackathon-october-2021}{\alert{Sign up form}}
\end{itemize}
\end{frame}
\end{document}
