<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8"/>
<title>Viral Integration</title>
<meta name="author" content="Edmund Miller"/>
<style type="text/css">
.underline { text-decoration: underline; }
</style>
<link rel="stylesheet" href="/home/emiller/.config/emacs/.local/straight/build-29.0.50/revealjs/dist/reveal.css"/>

<link rel="stylesheet" href="/home/emiller/.config/emacs/.local/straight/build-29.0.50/revealjs/dist/theme/white.css" id="theme"/>
</head>
<body>
<div class="reveal">
<div class="slides">
<section id="sec-title-slide">
<h1 class="title">Viral Integration</h1><h2 class="author">Edmund Miller</h2><h2 class="date">2022-07-05 Wed</h2><p class="date">Created: 2022-07-06 Wed 09:01</p>
</section>

<section>
<section id="slide-orgf7e3b52">
<h2 id="orgf7e3b52">nf-core/viralintegration</h2>
<div class="outline-text-2" id="text-orgf7e3b52">
</div>
</section>
<section id="slide-org3142e26">
<h3 id="org3142e26">Background</h3>
<ul>
<li>Started out wondering if we could look for Viral Integration in 1000 Genomes, inspired by <a href="http://oncodb.org/">OncoDB</a></li>
<li>Original: <code>bam2fastq</code> unaligned reads =&gt; nf-core/sarek (WGS) with a custom
reference</li>
<li>Met Robert Allaway, Principal Scientist @ Sage Bionetwork on nf-core</li>

</ul>

</section>
<section id="slide-org6ac9395">
<h3 id="org6ac9395">Background&#xa0;&#xa0;&#xa0;<span class="tag"><span class="ATTACH">ATTACH</span></span></h3>

<div id="org385761e" class="figure">
<p><img src="file:///home/emiller/src/personal/edmundmiller-dev/static/org-attach/b5/0d1404-f5b9-46d9-aeed-06210f093ad6/_20220706_090107screenshot.png" alt="_20220706_090107screenshot.png" />
</p>
</div>


</section>
<section id="slide-org632fbda">
<h3 id="org632fbda">Approaches</h3>
<ul>
<li>MetaPhlAn (Robert)</li>
<li>CTAT - Virus Integration Finder (ctat-VIF) (Alyssa Briggs)</li>
<li>BLASTBox (Edmund)</li>

</ul>

</section>
<section id="slide-orga63ec22">
<h3 id="orga63ec22">Why are we interested in Viral Integration?</h3>
<ul>
<li>Besides aberrant DNA methylation or RNA expression, another major cause of
cancer is viral infection (Oncoviruses)</li>
<li>Identification of oncovirus-related gene expression changes helps better
understand the mechanisms underlying virus-induced cancers</li>
<li>Lack of current research in the area utilizing new data</li>

</ul>

</section>
</section>
<section>
<section id="slide-org61fc580">
<h2 id="org61fc580">CTAT - Virus Integration Finder</h2>
<div class="outline-text-2" id="text-org61fc580">
</div>
</section>
<section id="slide-org20793de">
<h3 id="org20793de">CTAT - Virus Integration Finder</h3>
<ul>
<li>Capture evidence of virus-matching reads (consistent with an infection)</li>
<li>Identify and quantify evidence for viral insertion sites in the human genome</li>
<li>Provide interactive visualizations for evidence of virus-mapped reads and
virus insertion sites</li>

</ul>

</section>
<section id="slide-org83d450e">
<h3 id="org83d450e">CTAT - Virus Integration Finder</h3>
<ul>
<li>Trinity Cancer Transcriptome Analysis Toolkit - CWL</li>
<li>Takes DNA-seq or RNA-seq</li>
<li>Requires a CTAT genome lib supplemented with viral genomes (They provide a virusdb fasta).</li>

</ul>

</section>
<section id="slide-org20ae0db">
<h3 id="org20ae0db">CTAT - DAG&#xa0;&#xa0;&#xa0;<span class="tag"><span class="ATTACH">ATTACH</span></span></h3>

<div id="org0f252c5" class="figure">
<p><img src="file:///home/emiller/src/personal/edmundmiller-dev/static/org-attach/de/1b3c29-d7ca-4e44-8708-c72053d52682/_20220706_081414screenshot.png" alt="_20220706_081414screenshot.png" />
</p>
</div>


</section>
<section id="slide-orge7b0fd9">
<h3 id="orge7b0fd9">CTAT - Virus Integration Finder</h3>
<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">


<colgroup>
<col  class="org-left" />

<col  class="org-right" />

<col  class="org-left" />

<col  class="org-left" />

<col  class="org-right" />

<col  class="org-left" />

<col  class="org-left" />

<col  class="org-right" />

<col  class="org-right" />

<col  class="org-right" />

<col  class="org-right" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-left">chrA</th>
<th scope="col" class="org-right">coordA</th>
<th scope="col" class="org-left">orientA</th>
<th scope="col" class="org-left">chrB</th>
<th scope="col" class="org-right">coordB</th>
<th scope="col" class="org-left">orientB</th>
<th scope="col" class="org-left">prelim.primary<sub>brkpt</sub><sub>type</sub></th>
<th scope="col" class="org-right">prelim.total</th>
<th scope="col" class="org-right">split</th>
<th scope="col" class="org-right">span</th>
<th scope="col" class="org-right">total</th>
</tr>
</thead>
<tbody>
<tr>
<td class="org-left">chr11</td>
<td class="org-right">117344940</td>
<td class="org-left">+</td>
<td class="org-left">HPV16</td>
<td class="org-right">1215</td>
<td class="org-left">+</td>
<td class="org-left">Split</td>
<td class="org-right">117</td>
<td class="org-right">33</td>
<td class="org-right">79</td>
<td class="org-right">112</td>
</tr>

<tr>
<td class="org-left">chr18</td>
<td class="org-right">73320805</td>
<td class="org-left">+</td>
<td class="org-left">HPV16</td>
<td class="org-right">7409</td>
<td class="org-left">-</td>
<td class="org-left">Split</td>
<td class="org-right">168</td>
<td class="org-right">31</td>
<td class="org-right">72</td>
<td class="org-right">103</td>
</tr>

<tr>
<td class="org-left">chr6</td>
<td class="org-right">64935774</td>
<td class="org-left">+</td>
<td class="org-left">HPV16</td>
<td class="org-right">4034</td>
<td class="org-left">-</td>
<td class="org-left">Split</td>
<td class="org-right">177</td>
<td class="org-right">30</td>
<td class="org-right">70</td>
<td class="org-right">100</td>
</tr>

<tr>
<td class="org-left">HPV16</td>
<td class="org-right">4029</td>
<td class="org-left">+</td>
<td class="org-left">chr3</td>
<td class="org-right">937922</td>
<td class="org-left">+</td>
<td class="org-left">Split</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
</tr>

<tr>
<td class="org-left">HPV16</td>
<td class="org-right">1215</td>
<td class="org-left">-</td>
<td class="org-left">chr11</td>
<td class="org-right">108577428</td>
<td class="org-left">+</td>
<td class="org-left">Split</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
</tr>

<tr>
<td class="org-left">chr2</td>
<td class="org-right">32018732</td>
<td class="org-left">+</td>
<td class="org-left">HPV16</td>
<td class="org-right">1215</td>
<td class="org-left">+</td>
<td class="org-left">Split</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
</tr>

<tr>
<td class="org-left">chr1</td>
<td class="org-right">218562612</td>
<td class="org-left">+</td>
<td class="org-left">HPV16</td>
<td class="org-right">1215</td>
<td class="org-left">+</td>
<td class="org-left">Split</td>
<td class="org-right">1</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
<td class="org-right">0</td>
</tr>
</tbody>
</table>


</section>
<section id="slide-orgf3416d3">
<h3 id="orgf3416d3">CTAT - Insertion Site Candidates&#xa0;&#xa0;&#xa0;<span class="tag"><span class="ATTACH">ATTACH</span></span></h3>

<div id="org98c73e7" class="figure">
<p><img src="file:///home/emiller/src/personal/edmundmiller-dev/static/org-attach/0f/a01f6e-0bdc-49a0-8083-8e10c011cdb9/_20220705_222615screenshot.png" alt="_20220705_222615screenshot.png" />
</p>
</div>


</section>
<section id="slide-org382b615">
<h3 id="org382b615">CTAT - Virus Insertion Viewer&#xa0;&#xa0;&#xa0;<span class="tag"><span class="ATTACH">ATTACH</span></span></h3>

<div id="org08fe979" class="figure">
<p><img src="file:///home/emiller/src/personal/edmundmiller-dev/static/org-attach/66/211082-d11c-4fd4-bbbd-00d997ec7f4d/_20220705_223158screenshot.png" alt="_20220705_223158screenshot.png" />
</p>
</div>

</section>
<section id="slide-org3216a7d">
<h3 id="org3216a7d">CTAT - Virus Infection Evidence Viewer</h3>
<ul>
<li>There may not be strong evidence for virus insertions</li>
<li>Detects reads aligning to the target virus sequence</li>

</ul>

</section>
<section id="slide-orgcb993c8">
<h3 id="orgcb993c8">CTAT - Virus Infection Evidence Viewer&#xa0;&#xa0;&#xa0;<span class="tag"><span class="ATTACH">ATTACH</span></span></h3>

<div id="orgfb0a72e" class="figure">
<p><img src="file:///home/emiller/src/personal/edmundmiller-dev/static/org-attach/85/fec50f-b0cf-4721-98d0-3b1e5e56ba79/_20220706_085548screenshot.png" alt="_20220706_085548screenshot.png" />
</p>
</div>


</section>
</section>
<section>
<section id="slide-orgb4260a3">
<h2 id="orgb4260a3">BLASTBox</h2>
<div class="outline-text-2" id="text-orgb4260a3">
</div>
</section>
<section id="slide-org26bb5a0">
<h3 id="org26bb5a0">BLASTBox</h3>
<ul>
<li>Main goal was to <b>avoid realigning</b> thousands of high-coverage samples, by
pulling out unaligned reads</li>
<li>Found a similar study of TCGA data</li>

</ul>

</section>
<section id="slide-orgae8d883">
<h3 id="orgae8d883">BLASTBox - DAG&#xa0;&#xa0;&#xa0;<span class="tag"><span class="ATTACH">ATTACH</span></span></h3>

<div id="org9176d9d" class="figure">
<p><img src="file:///home/emiller/src/personal/edmundmiller-dev/static/org-attach/8a/152476-d37a-4dcf-a5ba-c698b37bf2d6/_20220706_082010screenshot.png" alt="_20220706_082010screenshot.png" />
</p>
</div>

</section>
<section id="slide-org491bd3d">
<h3 id="org491bd3d">BLASTBox&#xa0;&#xa0;&#xa0;<span class="tag"><span class="ATTACH">ATTACH</span></span></h3>

<div id="org4846025" class="figure">
<p><img src="file:///home/emiller/src/personal/edmundmiller-dev/static/org-attach/8d/1efa65-35ae-4394-95f6-a36ddd602529/_20220706_072610screenshot.png" alt="_20220706_072610screenshot.png" />
</p>
</div>

</section>
<section id="slide-org149374b">
<h3 id="org149374b">Viral expression associated with GIA Highlights (2013)</h3>
<ul>
<li>Goal: Determine whether the presence of a virus is significantly associated
with tumors</li>
<li>59% of screened gastrointestinal adenocarcinomas (GIA) were positive for at
least one virus</li>
<li>Used a similar approach, BAM =&gt; BLAST</li>

</ul>

</section>
<section id="slide-org6cd089e">
<h3 id="org6cd089e">BLASTBox Targets</h3>
<ul>
<li>1000 Genomes 30x WGS</li>
<li>TCGA WGS</li>
<li>dbGaP</li>

</ul>

</section>
<section id="slide-org638dc2d">
<h3 id="org638dc2d">Initial Questions to ask</h3>
<ul>
<li>What parts of the sequence got integrated?</li>
<li>What parts of the sequence got tossed out?</li>
<li>Which one is potentially toxic/oncogenic</li>

</ul>
</section>
</section>
</div>
</div>
<script src="/home/emiller/.config/emacs/.local/straight/build-29.0.50/revealjs/dist/reveal.js"></script>
<script src="/home/emiller/.config/emacs/.local/straight/build-29.0.50/revealjs/plugin/markdown/markdown.js"></script>
<script src="/home/emiller/.config/emacs/.local/straight/build-29.0.50/revealjs/plugin/notes/notes.js"></script>
<script src="/home/emiller/.config/emacs/.local/straight/build-29.0.50/revealjs/plugin/search/search.js"></script>
<script src="/home/emiller/.config/emacs/.local/straight/build-29.0.50/revealjs/plugin/zoom/zoom.js"></script>
<script>
// Full list of configuration options available here:
// https://github.com/hakimel/reveal.js#configuration
Reveal.initialize({

controls: true,
progress: true,
history: false,
center: true,
slideNumber: 'c',
rollingLinks: false,
keyboard: true,
mouseWheel: false,
fragmentInURL: false,
hashOneBasedIndex: false,
pdfSeparateFragments: true,
overview: true,

transition: 'convex',
transitionSpeed: 'default',

// Plugins with reveal.js 4.x
plugins: [ RevealMarkdown, RevealNotes, RevealSearch, RevealZoom ],

// Optional libraries used to extend reveal.js
dependencies: [
]

});
</script>
</body>
</html>
