% Created 2021-04-07 Wed 15:04
% Intended LaTeX compiler: pdflatex
\documentclass[bigger]{beamer}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{graphicx}
\usepackage{grffile}
\usepackage{longtable}
\usepackage{wrapfig}
\usepackage{rotating}
\usepackage[normalem]{ulem}
\usepackage{amsmath}
\usepackage{textcomp}
\usepackage{amssymb}
\usepackage{capt-of}
\usepackage{hyperref}
\usetheme[progressbar=foot]{metropolis}
\author{Edmund Miller}
\date{2021-04-07 Wed}
\title{Profiling of transcribed cis-regulatory elements in single cells}
\hypersetup{
 pdfauthor={Edmund Miller},
 pdftitle={Profiling of transcribed cis-regulatory elements in single cells},
 pdfkeywords={\href{../../concepts/biology/sequencing/chip\_seq.org}{ChIP-seq}},
 pdfsubject={a framework for deterministic machine learning},
 pdfcreator={Emacs 28.0.50 (Org mode 9.5)}, 
 pdflang={English}}
\begin{document}

\maketitle

\section{Background}
\label{sec:orgb7d8447}

\begin{frame}[label={sec:orgd54c556}]{CAGE-Seq}
\begin{itemize}
\item \href{https://pubmed.ncbi.nlm.nih.gov/19056941/}{GRO-seq - 2008 Dec}
\item \href{https://pubmed.ncbi.nlm.nih.gov/22362160/}{CAGE-Seq - 2012 Feb}
\begin{itemize}
\item Riken
\item Piero Carninci
\begin{itemize}
\item Leads the FANTOM project
\item Director of Riken Omics in 2008
\end{itemize}
\end{itemize}
\end{itemize}
\end{frame}

\begin{frame}[label={sec:orge00eecc}]{CAGE-Seq}
\begin{center}
\includegraphics[width=1.05\linewidth]{/home/emiller/sync/org/roam/data/41/05dc22-bb26-4c66-b02e-093aba53cb75/_20210407_122715screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:orga0f071e}]{CAGE-Seq}
\begin{block}{Pros:}
\begin{itemize}
\item Measures RNA expression levels and maps TSS in promoter regions
\item Provides precise mapping of TSS with single-nucleotide resolution
\end{itemize}
\end{block}

\begin{block}{Cons:}
\begin{itemize}
\item Only works on total mature RNA
\item Detection is biased toward TSS of long-lived transcripts
\end{itemize}
\end{block}
\end{frame}
\section{Introduction}
\label{sec:org357da36}

\begin{frame}[label={sec:org715f231}]{Overview of the experimental designs and benchmark analysis}
\begin{center}
\includegraphics[width=1.05\linewidth]{/home/emiller/sync/org/roam/data/41/05dc22-bb26-4c66-b02e-093aba53cb75/_20210407_133109screenshot.png}
\end{center}
\end{frame}
\begin{frame}[label={sec:org93544f0}]{tCRE and aCRE}
\begin{itemize}
\item tCREs are defined by merging closely located TSS clusters with +-500nt of gene TSS
\item aCREs are defined by the ATAC peak ranges
\end{itemize}
\end{frame}
\begin{frame}[label={sec:org513a1d4}]{SCAFE}
\begin{center}
\includegraphics[height=0.7\linewidth]{/home/emiller/sync/org/roam/data/41/05dc22-bb26-4c66-b02e-093aba53cb75/_20210407_144930screenshot.png}
\end{center}
\end{frame}


\section{Results}
\label{sec:orgafbb29d}
\begin{frame}[label={sec:org93497f2}]{Performance of sc-RNA-seq methods}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/41/05dc22-bb26-4c66-b02e-093aba53cb75/_20210407_134849screenshot.png}
\end{center}
\end{frame}


\begin{frame}[label={sec:orgceacdb0}]{Performance of sc-RNA-seq methods}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/41/05dc22-bb26-4c66-b02e-093aba53cb75/_20210407_141823screenshot.png}
\end{center}
\end{frame}
\begin{frame}[label={sec:orgd82d184}]{Definition and properties of tCRE}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/41/05dc22-bb26-4c66-b02e-093aba53cb75/_20210407_143847screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:org35e9b4c}]{Definition and properties of tCRE}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/41/05dc22-bb26-4c66-b02e-093aba53cb75/_20210407_143904screenshot.png}
\end{center}
\end{frame}
\begin{frame}[label={sec:org876254e}]{Results}
\begin{itemize}
\item Comparison of tCRE and aCRE in PBMCs
\item Disease-associated variants attCRE and aCRE in PBMCs
\end{itemize}
\end{frame}

\section{Discussion}
\label{sec:org4089da4}
\begin{frame}[label={sec:orgee7f9d2}]{Discussion}
\begin{itemize}
\item Can detect eRNAs with sc-end5-seq, however high level of dropouts
\begin{itemize}
\item Use of meta-cells might fix this
\item Alternative library prep with just nuclei or targeting eRNAs
\end{itemize}

\item sc-end5-seqdata can theoretically detect CRE activity with no extra cost
\begin{itemize}
\item Lack of dedicated tools for data analyses prevented the wider adoption
\end{itemize}
\end{itemize}
\end{frame}
\end{document}