% Created 2021-10-25 Mon 13:57
% Intended LaTeX compiler: pdflatex
\documentclass[bigger]{beamer}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{graphicx}
\usepackage{grffile}
\usepackage{longtable}
\usepackage{wrapfig}
\usepackage{rotating}
\usepackage[normalem]{ulem}
\usepackage{amsmath}
\usepackage{textcomp}
\usepackage{amssymb}
\usepackage{capt-of}
\usepackage{hyperref}
\usetheme[progressbar=foot]{metropolis}
\author{Edmund Miller}
\date{2021-10-25 Mon}
\title{Single-cell epigenomics reveals mechanisms of human cortical development}
\hypersetup{
 pdfauthor={Edmund Miller},
 pdftitle={Single-cell epigenomics reveals mechanisms of human cortical development},
 pdfkeywords={},
 pdfsubject={},
 pdfcreator={Emacs 28.0.50 (Org mode 9.5)}, 
 pdflang={English}}
\begin{document}

\maketitle

\section{Background}
\label{sec:orgeb6b205}
\begin{frame}[label={sec:org160b449}]{Chromatin Accessibility}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/98/fa6918-b773-42c5-986f-ecd891cba551/_20211025_113951screenshot.png}
\end{center}
\end{frame}
\begin{frame}[label={sec:orgeb31a7d}]{Enhancers}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/82/072fca-90bf-4ffb-8bbf-6f50cacd02d0/_20211025_114315screenshot.png}
\end{center}
\end{frame}


\begin{frame}[label={sec:orgc34be7d}]{Cerebral Organoids}
\begin{itemize}
\item Created \emph{in vitro} miniture organ resembling a brain
\begin{itemize}
\item Created by culturing pluripotent stem cells in a three-dimensional
rotational bioreactor
\item Chaotically arranged, instead of structural organized
\end{itemize}
\item Can be used to investigate early human brain development
\item Need to ensure the organoids form in a reproducible way
\end{itemize}
\end{frame}

\begin{frame}[label={sec:org53d587b}]{Motivation}
\begin{itemize}
\item Better understanding of signalling pathways that contribute to
neurodevelopmental delay (Vitamin A)
\item Better understanding of mammalian development
\item Provide a blueprint for evaluating cerebral organoids as a model for cortical
development
\end{itemize}
\end{frame}

\section{Methods}
\label{sec:orgaf2f66d}

\begin{frame}[label={sec:orgdd8c5e7}]{ATAC-seq}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/77/8b2dfa-d48f-4ab3-aa3a-d9c35b5e66c9/_20211025_094241nihms653929f1.jpg}
\end{center}

\begin{itemize}
\item A proxy to how easily a transcription factor can bind to the genome
\item Uses TN5 transposase, which binds to open chromatin and inserts DNA sequencing
adapters
\end{itemize}
\end{frame}


\begin{frame}[label={sec:orgfeb20db}]{Sequencing with Index/barcodes}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/04/478639-3acf-4f0a-8741-6680b4edf9da/_20211025_084832screenshot.png}
\end{center}
\end{frame}


\begin{frame}[label={sec:org5ba670e}]{Droplet-based cell capture}
\begin{center}
\includegraphics[height=0.45\linewidth]{/home/emiller/sync/org/roam/data/5c/d69f58-9ed6-49a0-b2a8-53b27727eac7/_20211025_084902screenshot.png}
\end{center}

\begin{itemize}
\item Cheaper per cell because of fewer doublets and ability to capture tens of thousands of cells
\item Popularized by Drop-seq and InDrop DIY systems
\item Commercially available platform is the 10x Genomics Chromium device
\end{itemize}
\end{frame}


\begin{frame}[label={sec:orgbd5a49b}]{Single Cell Sequencing}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/a6/fdba8a-777b-4a1a-b600-9feeda52c95c/_20211025_100129screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:org8466c6e}]{Fig 1b. Experimental Workflow}
\begin{center}
\includegraphics[width=1.05\linewidth]{/home/emiller/sync/org/roam/data/3f/a72657-d01a-43d9-86a7-21525d38d0f5/_20211024_160756screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:orgd35ea48}]{PLAC-seq}
\begin{center}
\includegraphics[height=0.4\linewidth]{/home/emiller/sync/org/roam/data/d3/29be93-bd88-4b1e-a925-0003711d2796/_20211024_173423screenshot.png}
\end{center}

\begin{itemize}
\item Looking for long-range chromatin loops
\item Rivaling Hi-C and ChIA-PET
\item Proximity ligation is conducted in nuclei prior to chromatin shearing and
immunoprecipitation (instead of proximity ligation are performed after
chromatin shearing)
\begin{itemize}
\item Ligats \alert{then} immunoprecipitates
\end{itemize}
\item PLAC-seq requires less input materials, and claims better accuracy than ChIA-PET
\end{itemize}
\end{frame}

\section{Results}
\label{sec:orgd8a259d}
\begin{frame}[label={sec:orgbbcd11b}]{Chromatin states of the developing brain}
\begin{itemize}
\item Performed scATAC-seq on primary samples of human forebrain mid-gestation
\item Confirmed that aggregate signal from the single-cell libraries matched up with
bulk ATAC-seq libraries created in parallel
\item Used CellWalker to assign cell-type labels based on previous scRNA-seq
\item Able to identify differentially accessible peaks between \alert{sub} types of cells
\end{itemize}
\end{frame}

\begin{frame}[label={sec:org3c202ae}]{Fig 1f. UMAP projection of primary scATAC-seq cells coloured by broad cell type}
\begin{center}
\includegraphics[height=0.55\linewidth]{/home/emiller/sync/org/roam/data/28/d496d0-8dbb-482c-8e5e-025378c68e32/_20211025_105356screenshot.png}
\end{center}
\end{frame}


\begin{frame}[label={sec:orgc42dec2}]{Identifying cell-type-specific enhancers}
\begin{itemize}
\item Intersected our peak set with the imputed 25-state chromatin model from Roadmap Epigenomics1
\item Identified cell-type-specific differentially accessible peaks for each cell
type
\item Overlaid with H3K27ac (Didn't use H3K4me1 because they only wanted active
enhancers?)
\item Used PLAC-seq and H3K4me3(promoter)
\end{itemize}
\end{frame}


\begin{frame}[label={sec:org003052e}]{Extended Data Fig. 5 - Venn diagram of overlap of all predicted enhancers}
\begin{center}
\includegraphics[height=0.6\linewidth]{/home/emiller/sync/org/roam/data/78/16b4e6-69ee-4a76-9df5-32665af5bf30/_20211024_224447screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:orgad1c216}]{Extended Data Fig. 5 - Venn diagram of overlap of all predicted enhancers}
\begin{center}
\includegraphics[height=0.6\linewidth]{/home/emiller/sync/org/roam/data/69/eaeb76-cb53-4d85-b4cd-599f176fc637/_20211025_121846screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:org149dab6}]{Fig 1h. predicted enhancer-gene interactions for RGs}
\begin{center}
\includegraphics[height=0.4\linewidth]{/home/emiller/sync/org/roam/data/87/16f37e-71e6-408d-8a93-364ffdb9bebb/_20211024_170901screenshot.png}
\end{center}

\begin{itemize}
\item \alert{Pink Curves} are predicted enhancer-gene interactions for RGs
\item \alert{Red Boxes} are predicted enhancers of ARX
\end{itemize}
\end{frame}

\begin{frame}[label={sec:org411305e}]{Fig 1h. LacZ staining regions of enhancer activity for enhancer candidates}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/32/7b075f-404f-46c6-bcc6-162ed22a087c/_20211024_171702screenshot.png}
\end{center}
\end{frame}


\begin{frame}[label={sec:org234d038}]{Fig 1g. Characterizing the regulatory 'grammar' of cell types}
\begin{center}
\includegraphics[height=0.5\linewidth]{/home/emiller/sync/org/roam/data/ee/77a47a-3315-4e62-9087-dcfdd53655c0/_20211025_104634screenshot.png}
\end{center}

\begin{itemize}
\item Working towards discovering transcription factor code that controls cell
lineage
\end{itemize}
\end{frame}

\begin{frame}[label={sec:orgfc4e140}]{Disease risk in the regulatory landscape}
\begin{itemize}
\item Explored mutations in non-coding genomic regions, and loss-of-function in
chromatin regulators
\item Linked ATAC-seq peaks, and putative enhancers with variants
\end{itemize}
\end{frame}

\begin{frame}[label={sec:org9ae2bc8}]{Fig 1ijk.}
\begin{center}
\includegraphics[height=0.5\linewidth]{/home/emiller/sync/org/roam/data/e3/5f7892-29a4-4cfa-b502-98bfea720abf/_20211025_110520screenshot.png}
\end{center}

\begin{itemize}
\item Potential to identify specific regulatory programs during cortical development
that confer the greatest risk for neurodevelopmental disorders
\end{itemize}
\end{frame}


\begin{frame}[label={sec:org3e3f0b9}]{Fig 2a Workflow for co-embedding scATAC-seq and scRNA-seq data}
\begin{center}
\includegraphics[height=0.45\linewidth]{/home/emiller/sync/org/roam/data/17/30c98e-88d8-4e5f-878a-37ac0e4b463f/_20211024_174146screenshot.png}
\end{center}

\begin{itemize}
\item Why 3n in scATAC-seq and 2n in scRNA-seq?
\item Right, UMAP projection of co-embedded scATAC-seq and scRNA-seq cells
coloured by Leiden clusters
\end{itemize}
\end{frame}

\begin{frame}[label={sec:orge4a3d30}]{Fig 2bc.}
\begin{center}
\includegraphics[height=0.5\linewidth]{/home/emiller/sync/org/roam/data/91/afebc0-c9b0-49da-b3f5-f7b63fd85d83/_20211025_110842screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:org21f471c}]{Fig 2e. Pseudotime}
\begin{center}
\includegraphics[height=0.7\linewidth]{/home/emiller/sync/org/roam/data/f1/a3cbe9-b2f6-443a-82a1-2f190d089e64/_20211025_111045screenshot.png}
\end{center}
\end{frame}


\begin{frame}[label={sec:orgf1dd2d5}]{Fig 2fgh. Enhancers predicted to interact with genes linked to cell type identity}
\begin{center}
\includegraphics[height=0.6\linewidth]{/home/emiller/sync/org/roam/data/38/1ecfce-c0a2-4cc8-8c59-f63152db0cb1/_20211025_111205screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:orgac8f13f}]{Gene activity score}
\begin{itemize}
\item a proxy for gene expression
\item ATACseq fragments in the gene body plus promoter (2 kb upstream from
transcription start sites) of all protein-coding genes were summed for each
cell
\end{itemize}
\end{frame}


\begin{frame}[label={sec:orgda96ccf}]{Fig 2i. Gene activity scores compared to gene expression}
\begin{center}
\includegraphics[height=0.6\linewidth]{/home/emiller/sync/org/roam/data/bc/9c8e8f-f0c0-4fe7-bc05-87f9e0dfc2ad/_20211025_111715screenshot.png}
\end{center}
\end{frame}
\begin{frame}[label={sec:orga4f51ad}]{Fig 2j. Dynamic chromatin states in neurogenesis}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/20/c57261-871c-4d86-9eea-50357d2b1240/_20211025_111855screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:org27abbc9}]{Fig 3a. Area-specific chromatin states}
\begin{center}
\includegraphics[height=0.6\linewidth]{/home/emiller/sync/org/roam/data/63/bbe34d-d025-4c13-bb3e-c1fef1aa42a0/_20211025_113028screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:orgd4ca7e4}]{Fig 3. Area-specific chromatin states}
\begin{center}
\includegraphics[height=0.5\linewidth]{/home/emiller/sync/org/roam/data/86/4f15f2-eb6a-46e7-93aa-8780ccd83187/_20211025_113105screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:org6b79ba6}]{Fig 3IM. Retinoic acid in cortical arealization}
\begin{center}
\includegraphics[height=0.7\linewidth]{/home/emiller/sync/org/roam/data/04/db7353-4613-4bcd-bf51-283c9aec6269/_20211025_113251screenshot.png}
\end{center}

\begin{itemize}
\item Retinoic acid signalling has an important role in patterning of the neural
tissue during mammalian brain development
\item Week 10 of differentiation corresponds to deep layer neurogenesis,
\item Neurons classified as PFC-like among organoids cultured with vitamin A
\end{itemize}
\end{frame}

\begin{frame}[label={sec:orgc0fd990}]{Benchmarking cerebral organoids}
\begin{center}
\includegraphics[height=0.6\linewidth]{/home/emiller/sync/org/roam/data/1d/5fcc47-3766-4345-bc7d-824ce2f45755/_20211025_115025screenshot.png}
\end{center}
\end{frame}


\begin{frame}[label={sec:orgff0fc54}]{Fig 4i. 80\% of predicted enhancers found in organiods}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/15/3e7149-61ba-4bac-877f-151d35f92cd2/_20211025_114927screenshot.png}
\end{center}

\begin{itemize}
\item Missing microglial enhancers
\end{itemize}
\end{frame}

\begin{frame}[label={sec:orgc1f3e16}]{Fig 4f. Overlap of Peaks}
\begin{center}
\includegraphics[height=0.5\linewidth]{/home/emiller/sync/org/roam/data/7c/03531f-ad7f-42ad-9b25-ccb51af2f31a/_20211025_114601screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:org7d40b67}]{Discussions}
\begin{itemize}
\item Found thousands of transiently accessible loci that track with neuronal
differentiation.
\item Found states that may reveal mechanisms for cell fate
\item Extend the established role of RA signaling in forebrain development
\begin{itemize}
\item RA signaling plays a role in the specificitation of excitatory neurons
\end{itemize}
\end{itemize}
\end{frame}

\begin{frame}[label={sec:org4767823}]{Article Critique}
\begin{itemize}
\item Small number of n
\item Difficulty reproducing enhancers
\end{itemize}
\end{frame}

\begin{frame}[label={sec:orgc55dd5c}]{Future Directions}
\begin{itemize}
\item Look for more disease-associated variants in regulatory regions that affect
the developing cortex
\end{itemize}
\end{frame}
\end{document}
