% Created 2021-12-08 Wed 09:49
% Intended LaTeX compiler: pdflatex
\documentclass[bigger]{beamer}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{graphicx}
\usepackage{grffile}
\usepackage{longtable}
\usepackage{wrapfig}
\usepackage{rotating}
\usepackage[normalem]{ulem}
\usepackage{amsmath}
\usepackage{textcomp}
\usepackage{amssymb}
\usepackage{capt-of}
\usepackage{hyperref}
\usetheme[progressbar=foot]{metropolis}
\author{Edmund Miller}
\date{2021-10-25 Mon}
\title{Nascent Transcript Identification Using CHM13}
\hypersetup{
 pdfauthor={Edmund Miller},
 pdftitle={Nascent Transcript Identification Using CHM13},
 pdfkeywords={},
 pdfsubject={},
 pdfcreator={Emacs 28.0.50 (Org mode 9.6)}, 
 pdflang={English}}
\begin{document}

\maketitle


\begin{frame}[label={sec:org4789d20}]{Cell Ranger}
\begin{itemize}
\item Wrapped up modules with test data
\item Pending addition to nf-core/scRNA-seq
\end{itemize}
\end{frame}


\begin{frame}[label={sec:org2a05d57}]{DRAGMAP}
\begin{itemize}
\item Competes with BWA-MEM
\item Pulled out of the DRAGEN-GATK pipeline
\item First version of DRAGMAP has not been optimized for speed.(1.10x-1.74x longer
than BWA)
\end{itemize}
\end{frame}

\begin{frame}[label={sec:org5262ac1}]{DRAGMAP}
\begin{center}
\includegraphics[height=0.5\linewidth]{/home/emiller/sync/org/roam/data/e8/91cfcb-aced-436c-bca6-7393d9d702fa/_20211207_203041screenshot.png}
\end{center}

\begin{itemize}
\item Fewer false positives
\end{itemize}
\end{frame}


\section{Kedro}
\label{sec:orgefd95dd}

\begin{frame}[label={sec:org88252df}]{Where we left off}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/38/5b5227-af9f-4353-8daa-1114d89464d2/_20211208_083410big_picture.png}
\end{center}
\end{frame}


\begin{frame}[label={sec:org2cf9ce6}]{Kedro}
\begin{itemize}
\item Kedro is an open-source Python framework for creating reproducible,
maintainable and modular data science code.
\item Data Catalog
\item Jupyter integration
\item Quick pipeline creation
\item Versioning
\item Testing
\end{itemize}
\end{frame}


\begin{frame}[label={sec:org99f9caa},fragile]{Kedro Data Catalog}
 \begin{verbatim}
bikes:
    type: pandas.CSVDataSet
    filepath: data/01_raw/bikes.csv

motorbikes:
    type: pandas.CSVDataSet
    filepath: s3://your_bucket/data/02_intermediate/company/motorbikes.csv
    credentials: dev_s3
    versioned: True
    load_args:
        sep: ','
        skiprows: 5
        na_values: ['#NA', NA]
\end{verbatim}
\end{frame}

\begin{frame}[label={sec:org0c297a9},fragile]{Usage}
 \begin{verbatim}
bikes = catalog.load("bikes")
motorbikes = catalog.load("motorbikes")
# pandas dataframe
bikes.head()
\end{verbatim}
\end{frame}

\section{nascent}
\label{sec:org7d7decc}

\begin{frame}[label={sec:orgb1749bb}]{Updates}
\begin{itemize}
\item groHMM works from any gtf(important for chm-13)
\item Update to new Native dsl2 syntax
\item Added DRAGMAP
\end{itemize}
\end{frame}

\begin{frame}[label={sec:orge432c97}]{Example Transcript Density Plot}
\begin{center}
\includegraphics[height=0.6\linewidth]{/home/emiller/sync/org/roam/data/9d/27e1c2-8a3c-48fc-a005-f25b1f316a66/_20211208_083630screenshot.png}
\end{center}
\end{frame}


\begin{frame}[label={sec:org3246047}]{CHM13}
\begin{itemize}
\item Used PacBio's HiFi, and Nanopore's ``ultra-long'' reads to resolve complex forms
of structural variation and gaps in GRCh38
\item CHM13 Cell line a complete hydatidiform mole (CHM) cell line, ``essentially
haploid nature''
\end{itemize}
\end{frame}

\begin{frame}[label={sec:orge17da0c}]{CHM13}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/43/29b3e4-e1bf-420b-a790-230e4949f9f5/_20211208_084146screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:orgd5d287e}]{CHM13}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/roam/data/cf/34f818-2d1d-4614-b608-cb5e26e328c0/_20211208_084238screenshot.png}
\end{center}
\end{frame}
\end{document}
