#+TITLE: nf-core & Selene
#+REVEAL_THEME: white
#+OPTIONS: num:nil
#+OPTIONS: toc:nil
#+OPTIONS: reveal_title_slide:nil
#+REVEAL_ROOT: https://cdn.jsdelivr.net/npm/reveal.js
#+REVEAL_HLEVEL: 2
* https://nf-co.re/assets/img/logo/nf-core-logo.svg
https://nf-co.re/
#+BEGIN_QUOTE
A community effort to collect a curated set of analysis pipelines built using Nextflow.
#+END_QUOTE
[[https://www.nature.com/articles/s41587-020-0439-x.epdf?shared_access_token=W-gHNzA4HngzDDSTRrfBRdRgN0jAjWel9jnR3ZoTv0P9zw2JZTis0qRZaQ0qtypN0g4rwhNebqVD1V-gt-AOlA82ul659wBl7hYkgj3IQbVny9QNsU0jdhfynPtglNcfpqimaGrEVJmM7gUo8H6lyg%3D%3D][Full Access Link]]
** Where we left off
#+BEGIN_SRC shell
cookiecutter gl:functional-genomics/cookiecutter-snakemake-workflow
#+END_SRC
** 
:PROPERTIES:
:reveal_background: ./nf-core.jpg
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
** Deploy
*** Getting started
#+BEGIN_SRC sh
# Install Nextflow
curl -fsSL get.nextflow.io | bash
# Add Nextflow binary to your user's PATH:
mv nextflow ~/bin/

nextflow run nf-core/rnaseq -profile docker \
    --singleEnd --reads '*.fastq' --genome GRCm38

nextflow run nf-core/rnaseq -profile utd_ganymede \
    --singleEnd --reads '*.fastq' --genome GRCm38
#+END_SRC
*** [[https://asciinema.org/a/EfVHOhIS3XC3nzWm0NwupLgGc][Demo]]
#+REVEAL_HTML: <a href="https://asciinema.org/a/EfVHOhIS3XC3nzWm0NwupLgGc" target="_blank"><img src="https://asciinema.org/a/EfVHOhIS3XC3nzWm0NwupLgGc.svg" height="550"/></a>
*** MK config coming soon!

** Participate
*** Documentation
*** [[https://nf-co.re/pipelines][Pipelines]]
- [[https://nf-co.re/rnaseq][RNA-Seq]]
- [[https://nf-co.re/scrnaseq][scRNA-Seq]]
- [[https://nf-co.re/atacseq][ATAC-Seq]]
- [[https://nf-co.re/deepvariant][Deep Variant]]
*** RNA-Seq
[[https://nf-co.re/rnaseq/docs/output#pipeline-overview][pipeline-overview]]
- fastqc/TrimGalore
- STAR/hisat/salmon
- featureCounts
- edgeR
- Multiqc
*** Forking
** Develop
*** [[https://nf-co.re/tools#creating-a-new-workflow][Creating a new workflow]]
#+BEGIN_SRC sh
pip install nf-core
nf-core create
#+END_SRC
*** [[https://asciinema.org/a/8VfGOnzYKPhK6nGubG9pBkHcx][Demo]]
#+REVEAL_HTML: <a href="https://asciinema.org/a/8VfGOnzYKPhK6nGubG9pBkHcx" target="_blank"><img src="https://asciinema.org/a/8VfGOnzYKPhK6nGubG9pBkHcx.svg" height="550"/></a>
* https://github.com/FunctionLab/selene/raw/master/docs/source/_static/img/selene_logo.png
** Links
- [[https://www.nature.com/articles/s41592-019-0360-8][Selene: a PyTorch-based deep learning library for sequence data | Nature Methods]]
- [[https://github.com/FunctionLab/selene][GitHub - FunctionLab/selene]]
** [[https://hb.flatironinstitute.org/deepsea/][DeepSEA]]
DeepSEA is a deep learning-based algorithmic framework for predicting the
chromatin effects of sequence alterations with single nucleotide sensitivity.
** Overview
- Trying to solve the issue of creating new deep learning architectures and
  training them on new data sets.
- Created to be a framework to be easy to use for deep learning in biology.
** Case 1: Training an exisiting architecture on a different dataset
- Trains DeepSEA module for transcription factor GATA1
- Zero lines of python code
** Case 2: Developing a new architecture and comparing performance across architectures
- Allows use of PyTorch modules
- Fully compatible with [[https://kipoi.org/][Kipoi]]
- Focus is on experimenting with various models rather than repeating code for
  training and evaluation.
** Case 3: Applying a new model to variants
- They train the new model from Case 2 to be used on the Alzheimer's disease
  GWAS to find varent effects that are significant and non-significant.
** Online methods
- Has an SDK and a CLI
- SDK is a python package for use in example with a web app to allow researchers
  to submit sequences.
* Learning Resources
- [[https://www.coursera.org/learn/sciwrite][Writing in the Science]]
- [[https://missing.csail.mit.edu/][The Missing Semester]]
- [[https://takesmartnotes.com/][How to Take Smart Notes]]
