#+title: Nascent Transcript Identification
#+AUTHOR:    Edmund Miller
#+EMAIL:     Edmund.Miller@utdallas.edu
#+DATE:      2022-03-02 Wed
#+DESCRIPTION:
#+KEYWORDS:
#+LANGUAGE:  en
#+OPTIONS:   H:2 num:t toc:nil \n:nil @:t ::t |:t ^:t -:t f:t *:t <:t
#+OPTIONS:   TeX:t LaTeX:t skip:nil d:nil todo:t pri:nil tags:not-in-toc
#+INFOJS_OPT: view:nil toc:nil ltoc:t mouse:underline buttons:0 path:https://orgmode.org/org-info.js
#+EXPORT_SELECT_TAGS: export
#+EXPORT_EXCLUDE_TAGS: noexport
#+HTML_LINK_UP:
#+HTML_LINK_HOME:
#+startup: beamer
#+LaTeX_CLASS: beamer
#+LaTeX_CLASS_OPTIONS: [bigger]
#+BEAMER_FRAME_LEVEL: 2
# C-c C-e l O
# SPC m e l O

* Background

** Previous Work :ATTACH:
:PROPERTIES:
:ID:       b6f0adf3-3b75-4179-9573-79ec2d6a4e52
:END:


[[attachment:_20220308_180148screenshot.png]]

** Experimental Design :ATTACH:
:PROPERTIES:
:ID:       dd512a5d-aabd-4e19-b91b-101bae668325
:END:


[[attachment:_20220308_180441screenshot.png]]


** eRNA Identification :ATTACH:
:PROPERTIES:
:ID:       1384c3c4-3022-40f7-ba62-d5eb11cbd95e
:END:


[[attachment:_20220308_180909screenshot.png]]

** Viral Inducible eRNA-gene pairs :ATTACH:
:PROPERTIES:
:ID:       d508c4a7-d758-4d71-b860-fd66a6e073ca
:END:


[[attachment:_20220308_181202screenshot.png]]


*** Inducibility criteria

#+attr_latex: :height 0.1\linewidth
[[attachment:_20220308_211437screenshot.png]]

[[attachment:_20220308_211445screenshot.png]]



** Reproduction with IMR Dataset

- Wrote workflow using snakemake
- Goal was to reproduce GM results
- Achieved 80% of predicted eRNAs

** GM12878/IMR90 Comparison :ATTACH:
:PROPERTIES:
:ID:       0d7b01dd-013f-4af3-a97d-96f5c407e5d1
:END:


[[attachment:_20220308_181956screenshot.png]]


* nf-core/nascent

** Standardizing Snakemake

- January 2020
- Template
- Universal Commands
- Testing
- CI/CD
- Wrappers

** nf-core Paper :ATTACH:
:PROPERTIES:
:ID:       451876b4-d9c8-47e7-a0e4-76e3e6403a4a
:END:

[[attachment:_20210127_123601screenshot.png]]

** nf-core Features :ATTACH:
:PROPERTIES:
:ID:       b9be7b67-b57f-4f58-8cda-36455fb83c53
:END:

#+attr_latex: :height 0.7\linewidth
[[attachment:_20210127_123835screenshot.png]]

** nf-core Getting started

#+attr_latex: :height 0.7\linewidth
#+begin_src bash
# Install nextflow
curl -s https://get.nextflow.io | bash
mv nextflow ~/bin/

# Launch the RNAseq pipeline
nextflow run nf-core/rnaseq \
    --input samplesheet.csv \
    --genome GRCh37 \
    -profile docker
#+end_src



** Inheriting nf-core Nascent

- Breaking our analysis up into smaller pieces
- nf-core portion includes Quality Checks, alignment, graph generation, transcript
  identification, and transcript quantification
- Downstream analysis is then a seperate nextflow workflow
- Data engineering/Data Science split

* Future
** Future

- CHM13 centromeres
- Utilizing PINTs transcriptional regulatory elements (TREs)
- WGS viral sequences

** CHM13

- Used PacBio's HiFi, and Nanopore's "ultra-long" reads to resolve complex forms
  of structural variation and gaps in GRCh38
- CHM13 Cell line a complete hydatidiform mole (CHM) cell line, "essentially
  haploid nature"

** Pacbio HiFi Reads :ATTACH:
:PROPERTIES:
:ID:       d253f799-3eb8-443c-9d93-7084900a978c
:END:



#+attr_latex: :height 0.4\linewidth
[[attachment:_20220302_085317screenshot.png]]



** CHM13 :ATTACH:
:PROPERTIES:
:ID:       4329b3e4-e1bf-420b-a790-230e4949f9f5
:END:

[[attachment:_20211208_084146screenshot.png]]

** CHM13 :ATTACH:
:PROPERTIES:
:ID:       cf34f818-2d1d-4614-b608-cb5e26e328c0
:END:

[[attachment:_20211208_084238screenshot.png]]


** The intersection of Homer identified peaks and centromeres

- Near identical number of reads mapped(There's only 5% more to align to)
- Had to use hg19 centromeres for the intersection(Found a T2T liftOver
  chainfile)
- There were no hits
- Confirmed with call to aligned reads for centromere regions

** The intersection of Homer identified peaks and centromeres

- Reads are probably being thrown out for two reasons
  - Centromeres are high in repeats so tossed
  - They then are high in repeats so the probably have multiple alignments and
    get tossed

** PINTS - transcriptional regulatory elements (TREs) :ATTACH:
:PROPERTIES:
:ID:       1ed9c9e3-b1d8-4fa0-833b-b387f82689bb
:END:

- PINTS is modeled after MACS2
- Utilized TSS assays (GRO/PRO-*cap*, CoPRO, Start-seq, CAGE)
- GRO/PRO-*seq* are NT assays


[[attachment:_20220308_205916screenshot.png]]


** Picking up Nascent Reads :ATTACH:
:PROPERTIES:
:ID:       e8130625-84b2-4d0f-8a4b-525b7459c08d
:END:

#+attr_latex: :height 0.7\linewidth
[[attachment:_20220308_210616screenshot.png]]

** PINTS Highlights

- Reasonable computational requirements (Feasible to run on a typical personal
  computer)
- Noticed *minor changes* in sample processing could lead to changes of up to
  *>20%* in the final results
- Well packaged and usable by others

** WGS viral sequences

- Take 1000 genomes data from Illumina
- Take unaligned reads from bams and convert back to raw reads
- Realign to viral genomes and quantify through nf-core/sarek WGS pipline
- Looking for possible correlation between viral infection and phenotypes

** Call to Actions

- Take [[https://functional-genomics-lab.github.io/Applied-Genomics/][*Applied Genomics*]] this summer!
- Apply for the [[https://nf-co.re/mentorships][*nf-core mentorship program*]]!
- [[https://registration.elementbiosciences.com/][*Element Biosciences launch event on monday*]]!