:PROPERTIES:
:ID:       bc34f638-403e-4879-a27e-dba787051972
:END:
#+title: Journal Club
#+author: Edmund Miller
#+email: Edmund.Miller@utdallas.edu
#+date:  2020-10-21 Wed
#+description:
#+keywords:
#+language:  en
#+options:   H:3 num:t toc:nil \n:nil @:t ::t |:t ^:t -:t f:t *:t <:t
#+options:   TeX:t LaTeX:t skip:nil d:nil todo:t pri:nil tags:not-in-toc
#+infojs_opt: view:nil toc:nil ltoc:t mouse:underline buttons:0 path:https://orgmode.org/org-info.js
#+export_select_tags: export
#+export_exclude_tags: noexport
#+link_up:
#+link_home:
#+startup: beamer
#+LaTeX_CLASS: beamer
#+LaTeX_CLASS_OPTIONS: [bigger]
#+beamer_frame_level: 2
# C-c C-e l O

* Roam Update
* Total functional score of enhancer elements identifies lineage-specific enhancers that drive differentiation of pancreatic cells
** Total Functional Score of Enhancer Elements (TFSEE) Model
*** Fig 1. :ATTACH:
:PROPERTIES:
:ID:       7123668f-b3d0-4eb9-8bf7-be04303f495b
:END:

#+attr_latex: :height 0.7\linewidth
[[attachment:_20201021_110044screenshot.png]]

*** Fig. S2A Pt1 Unbiased, genome-wide prediction of active enhancers :ATTACH:
:PROPERTIES:
:ID:       387ae15d-3704-4085-b1c0-040ccea415ac
:END:

#+attr_latex: :height 0.7\linewidth
[[attachment:_20201021_112515screenshot.png]]

*** Fig. S2A Pt2 Unbiased, genome-wide prediction of active enhancers :ATTACH:
:PROPERTIES:
:ID:       be3045b4-88aa-419c-8d6b-a77ab568cac0
:END:

#+attr_latex: :height 0.7\linewidth
[[attachment:_20201021_112522screenshot.png]]

*** Fig. S2B Unbiased, genome-wide prediction of active enhancers :ATTACH:
:PROPERTIES:
:ID:       e760b6ba-a9c3-4875-8a1e-5f8c91f887e7
:END:

#+attr_latex: :height 0.6\linewidth
[[attachment:_20201021_111014screenshot.png]]
*** Fig 2. Data Processing for Total Functional Score of Enhancer Elements (TFSEE) Method :ATTACH:
:PROPERTIES:
:ID:       bf933f7d-1b14-4a48-b275-f971fd8a8767
:END:

[[attachment:_20201021_121904screenshot.png]]
*** Fig 3. Overview of Total Functional Score of Enhancer Elements (TFSEE) Method :ATTACH:
:PROPERTIES:
:ID:       9a101d7f-896d-4ddb-add1-ba78abda35d7
:END:

[[attachment:_20201021_123427screenshot.png]]
*** Fig 4A. :ATTACH:
:PROPERTIES:
:ID:       2781b238-98e6-4771-8aad-2e1a5ad84731
:END:

#+attr_latex: :height 0.7\linewidth
[[attachment:_20201021_130023screenshot.png]]

Top) Schematic of pancreatic differentiation starting from Human embryonic stem cells (hESCs) to pancreatic endoderm (PE). (Bottom) Depiction of epigenomic (ChIP-seq) and transcriptional (GRO-seq and RNA-seq) profiles for each cell line used for analysis.

*** Fig 4B. :ATTACH:
:PROPERTIES:
:ID:       365e2d3c-9a61-42e5-939b-4d3d99a76570
:END:

#+attr_latex: :height 0.7\linewidth
[[attachment:_20201021_130039screenshot.png]]

*** Fig 4C. :ATTACH:
:PROPERTIES:
:ID:       ffc8ec28-63e1-49db-a3b3-3a53446c6f8b
:END:

#+attr_latex: :height 0.7\linewidth
[[attachment:_20201021_130054screenshot.png]]

*** Fig 4D. :ATTACH:
:PROPERTIES:
:ID:       448ae15d-e1f1-4e23-8753-6622897ef1a3
:END:

#+attr_latex: :height 0.7\linewidth
[[attachment:_20201021_130111screenshot.png]]

** TFSEE identifies lineage-specific enhancers and their cognate TFs during pancreatic differentiation
*** Figure 5. TFSEE identifies cell type-specific enhancers and their cognate TFs that drive gene expression during pancreatic differentiation :ATTACH:
:PROPERTIES:
:ID:       4c4f97f6-5aef-4f41-927c-fd8680640530
:END:

#+attr_latex: :height 0.6\linewidth
[[attachment:_20201021_131014screenshot.png]]
*** Figure 6. TFSEE-predicted TFs are enriched in pre and late pancreatic differentiation :ATTACH:
:PROPERTIES:
:ID:       d28dfe2a-7408-42a6-93e6-c369d67b0467
:END:

#+attr_latex: :height 0.7\linewidth
[[attachment:_20201021_132934screenshot.png]]

** Discussion
*** Strengths and Limitations
- TFSEE enables analysis of driver TFs using a limited amount of data
- The TFSEE model was able to identify lineage-specific TFs with as little as 5
  cell types and with only 2 data types, RNA-seq and ChIP-seq (for H3K4me3,
  H3K4me1, and H3K27ac)
- A limitation of the TFSEE method is that while the model can be used with a
  reduced number of data types for enhancer identification, it fails to identify
  additional subtype- or stage-specific drivers with reduced data input
*** Integrating additional genomic data into TFSEE
- Integrate genomic data indicating open regions of chromatin (ATAC-seq,
  DNase-seq, or MNase-seq)
- ChromHMM could be used to annotate alternate chromatin states with additional
  histone modifications
- Chromatin Looping data for enhancer-promoter interactions (as measured by 4C,
  ChIA-PET, or Hi-C)
