% Created 2020-10-21 Wed 13:56
% Intended LaTeX compiler: pdflatex
\documentclass[bigger]{beamer}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{graphicx}
\usepackage{grffile}
\usepackage{longtable}
\usepackage{wrapfig}
\usepackage{rotating}
\usepackage[normalem]{ulem}
\usepackage{amsmath}
\usepackage{textcomp}
\usepackage{amssymb}
\usepackage{capt-of}
\usepackage{hyperref}
\usetheme[progressbar=foot]{metropolis}
\author{Edmund Miller}
\date{2020-10-21 Wed}
\title{Journal Club}
\hypersetup{
 pdfauthor={Edmund Miller},
 pdftitle={Journal Club},
 pdfkeywords={},
 pdfsubject={},
 pdfcreator={Emacs 28.0.50 (Org mode 9.4)}, 
 pdflang={English}}
\begin{document}

\maketitle
\section{Roam Update}
\label{sec:org07d4d71}
\section{Total functional score of enhancer elements identifies lineage-specific enhancers that drive differentiation of pancreatic cells}
\label{sec:org1786a88}
\subsection{Total Functional Score of Enhancer Elements (TFSEE) Model}
\label{sec:orgcdaec4e}
\begin{frame}[label={sec:orgfadfdec}]{Fig 1.}
\begin{center}
\includegraphics[height=0.7\linewidth]{/home/emiller/sync/org/.attach/71/23668f-b3d0-4eb9-8bf7-be04303f495b/_20201021_110044screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:org638ff74}]{Fig. S2A Pt1 Unbiased, genome-wide prediction of active enhancers}
\begin{center}
\includegraphics[height=0.7\linewidth]{/home/emiller/sync/org/.attach/38/7ae15d-3704-4085-b1c0-040ccea415ac/_20201021_112515screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:org8e4cc61}]{Fig. S2A Pt2 Unbiased, genome-wide prediction of active enhancers}
\begin{center}
\includegraphics[height=0.7\linewidth]{/home/emiller/sync/org/.attach/be/3045b4-88aa-419c-8d6b-a77ab568cac0/_20201021_112522screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:org1781b03}]{Fig. S2B Unbiased, genome-wide prediction of active enhancers}
\begin{center}
\includegraphics[height=0.6\linewidth]{/home/emiller/sync/org/.attach/e7/60b6ba-a9c3-4875-8a1e-5f8c91f887e7/_20201021_111014screenshot.png}
\end{center}
\end{frame}
\begin{frame}[label={sec:org685e586}]{Fig 2. Data Processing for Total Functional Score of Enhancer Elements (TFSEE) Method}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/.attach/bf/933f7d-1b14-4a48-b275-f971fd8a8767/_20201021_121904screenshot.png}
\end{center}
\end{frame}
\begin{frame}[label={sec:orga33a8b0}]{Fig 3. Overview of Total Functional Score of Enhancer Elements (TFSEE) Method}
\begin{center}
\includegraphics[width=.9\linewidth]{/home/emiller/sync/org/.attach/9a/101d7f-896d-4ddb-add1-ba78abda35d7/_20201021_123427screenshot.png}
\end{center}
\end{frame}
\begin{frame}[label={sec:orgd83e7e4}]{Fig 4A.}
\begin{center}
\includegraphics[height=0.7\linewidth]{/home/emiller/sync/org/.attach/27/81b238-98e6-4771-8aad-2e1a5ad84731/_20201021_130023screenshot.png}
\end{center}

Top) Schematic of pancreatic differentiation starting from Human embryonic stem cells (hESCs) to pancreatic endoderm (PE). (Bottom) Depiction of epigenomic (ChIP-seq) and transcriptional (GRO-seq and RNA-seq) profiles for each cell line used for analysis.
\end{frame}

\begin{frame}[label={sec:org56fadb5}]{Fig 4B.}
\begin{center}
\includegraphics[height=0.7\linewidth]{/home/emiller/sync/org/.attach/36/5e2d3c-9a61-42e5-939b-4d3d99a76570/_20201021_130039screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:orgdb3d26c}]{Fig 4C.}
\begin{center}
\includegraphics[height=0.7\linewidth]{/home/emiller/sync/org/.attach/ff/c8ec28-63e1-49db-a3b3-3a53446c6f8b/_20201021_130054screenshot.png}
\end{center}
\end{frame}

\begin{frame}[label={sec:orgf89b08d}]{Fig 4D.}
\begin{center}
\includegraphics[height=0.7\linewidth]{/home/emiller/sync/org/.attach/44/8ae15d-e1f1-4e23-8753-6622897ef1a3/_20201021_130111screenshot.png}
\end{center}
\end{frame}

\subsection{TFSEE identifies lineage-specific enhancers and their cognate TFs during pancreatic differentiation}
\label{sec:orga8beff8}
\begin{frame}[label={sec:orgffe2a47}]{Figure 5. TFSEE identifies cell typeâ€“specific enhancers and their cognate TFs that drive gene expression during pancreatic differentiation}
\begin{center}
\includegraphics[height=0.6\linewidth]{/home/emiller/sync/org/.attach/4c/4f97f6-5aef-4f41-927c-fd8680640530/_20201021_131014screenshot.png}
\end{center}
\end{frame}
\begin{frame}[label={sec:org27a5dcc}]{Figure 6. TFSEE-predicted TFs are enriched in pre and late pancreatic differentiation}
\begin{center}
\includegraphics[height=0.7\linewidth]{/home/emiller/sync/org/.attach/d2/8dfe2a-7408-42a6-93e6-c369d67b0467/_20201021_132934screenshot.png}
\end{center}
\end{frame}

\subsection{Discussion}
\label{sec:org6a3c489}
\begin{frame}[label={sec:orgf7f1ed4}]{Strengths and Limitations}
\begin{itemize}
\item TFSEE enables analysis of driver TFs using a limited amount of data
\item The TFSEE model was able to identify lineage-specific TFs with as little as 5
cell types and with only 2 data types, RNA-seq and ChIP-seq (for H3K4me3,
H3K4me1, and H3K27ac)
\item A limitation of the TFSEE method is that while the model can be used with a
reduced number of data types for enhancer identification, it fails to identify
additional subtype- or stage-specific drivers with reducde data input
\end{itemize}
\end{frame}
\begin{frame}[label={sec:orga915143}]{Integrating additional genomic data into TFSEE}
\begin{itemize}
\item Integrate genomic data indicating open regions of chromatin (ATAC-seq,
DNase-seq, or MNase-seq)
\item ChromHMM could be used to annotate alternate chromatin states with additional
histone modifications
\item Chromatin Looping data for enhancer-promoter interactions (as measured by 4C,
ChIA-PET, or Hi-C)
\end{itemize}
\end{frame}
\end{document}
