#+title: Nascent Transcript Identification
#+AUTHOR:    Edmund Miller
#+EMAIL:     Edmund.Miller@utdallas.edu
#+DATE:      2022-09-07 Wed
#+reveal_theme: white
#+options: num:nil
#+options: toc:nil
#+options: reveal_title_slide:nil
#+startup: inlineimages:nil
#+startup: beamer
#+LaTeX_CLASS: beamer
#+LaTeX_CLASS_OPTIONS: [bigger]
#+BEAMER_FRAME_LEVEL: 2
# SPC m e l O


** groHMM fix
- Failed whenever we ran on full datasets
- Sruthi fixed it by adding ~keepStandardChromosomes~
- I expect this to have issues with CHM13
* CHM13 Struggles
** CHM13 Struggles

- v2.0 has been released
- genbank chr aliases are used by default
| genbank    | refseq       | assembly | ncbi | ucsc  |
|------------+--------------+----------+------+-------|
| CP068255.2 | NC\_060947.1 | X        | X    | chrX  |
| CP068256.2 | NC\_060946.1 | 22       | 22   | chr22 |

** CHM13 Struggles :ATTACH:
:PROPERTIES:
:ID:       996c3081-dce2-41c5-bccc-8de86c58e920
:END:


[[attachment:_20220906_220028screenshot.png]]
** CHM13 Struggles

- Rebuilding indexes with refgenie
- In process of getting them hosted to AWS igenomes for ease of use.

* Understanding PINTS Results
** IOU of GENCODE (Annot.) and those predicted by different tools :ATTACH:
:PROPERTIES:
:ID:       dcf0d0a6-1ba7-4b67-97e5-74b9d4ca892d
:END:

[[attachment:_20220907_081721screenshot.png]]

Consistencies vary greatly between transcription units annotated

** Schematic plot of PINTS :ATTACH:
:PROPERTIES:
:ID:       5233a535-b99a-4e1e-831e-63fa79c3e921
:END:


[[attachment:_20220907_082129screenshot.png]]


** How to count PINTS TREs?

1. _1 to 1 transcripts to samples_. The predicted TSS's for IMR0h would only be
   counted for IMR0h, and not counted for IMR1h. *One TSV per sample*
2. _Transcripts are counted across all samples_. So the TSS's for IMR0h are
   counted across all the samples and then combined. So you'd end up with a *tsv
   per sample x sample* with IMR0h TSS across all the samples, a TSV for IMR1h.
3. _Combine the TSS's across all samples and count once per sample_. end up with
   *one TSV* with all the transcripts counts for every sample.(What happens with
   homer/groHMM)

** How to count PINTS TREs?

#+begin_quote
For some types of analysis, such as transcript identification, it is a good idea
to create a single META-experiment that contains all of the GRO-Seq reads for a
given cell type
#+end_quote

Planning on something similar to *option 3* to follow homer recommendations to
avoid missing low transcription

** Future action

- Couple of outstanding GitHub issues on PINTS
- Updating test dataset to have at-least a "peak"(artificially selected)
- Create a test dataset that uses chr 21 for sample 1 and chr 22 for sample 2
  and see if there's any cross-over

* How do we determine 3' end?
** PINC

- Prediction of lncRNA based on RNA-Seq data.
- Easier problem because they can filter based off the coding-potential
- Brought some inspiration for basic filtering for end-users

** PINTS Issues

- Are we getting just the TSS with the GROSeq setting? Or the full nascent transcript?
- If it's just the TSS, could we combine their improved TSS identification to
  filter homer or groHMM /de novo/ transcripts?

** Homer Identification :ATTACH:
:PROPERTIES:
:ID:       a26ed570-97a2-4f5f-9c9c-72b5857c63d5
:END:

[[attachment:_20220907_071614screenshot.png]]

** PINTS for refining nascent transcripts :ATTACH:
:PROPERTIES:
:ID:       6a4a290f-4071-49dd-abdf-bfa0beaaf15d
:END:


[[attachment:_20220907_075231screenshot.png]]

** Training a model to find 3' ends

- Shayne mentioned she still does a lot of manual validation
- Perhaps something similar Deepvariant that "looks" at the pileups
